Emilie's script for metagenomics by organism
========================================================
For the python package Qiime:
Set directories and variables:
```{r}

analysisFolder="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU"
original_its="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/tsv_tables_original/its"
original_om="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/tsv_tables_original/om"
original_phy="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/tsv_tables_original/phy"
out_biom_ITS="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/BIOM/its"
out_biom_Om="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/BIOM/om"
out_biom_Phy="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/BIOM/phy"
ITS_merged_biom="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/ITS_merged/biom"
Om_merged_biom="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/Om_merged/biom"
Phy_merged_biom="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/Phy_merged/biom"
ITS_merged_tsv="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/ITS_merged/tsv"
Om_merged_tsv="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/Om_merged/tsv"
Phy_merged_tsv="/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/merge_OTU/Phy_merged/tsv"
qiimePathbiomformat="/isilon/biodiversity/pipelines/qiime-1.7.0/biom-format-1.1.2-release/bin/"
activateQiime <- "/isilon/biodiversity/pipelines/qiime-1.7.0/activate.sh"
qiimePath <- "/isilon/biodiversity/pipelines/qiime-1.7.0/qiime-1.7.0-release/bin"
ScriptsPath <- "/home/CFIA-ACIA/tremblaye/scripts/"

sourceWaitQiime <- paste(" source ", activateQiime, " && ", sep = "")
assignTaxQiime <- paste(qiimePath, "/assign_taxonomy.py", sep = "")


cmd <- paste(sourceWaitQiime, assignTaxQiime, " -h ", sep = "")
system(cmd)



###################################################
#                                                 #
#       Convert .tsv tables in biom tables        #
#                                                 #
###################################################

##############
#            #
#    ITS     #
#            #
##############


for i in ${original_its}/*.tsv; do 
    name="$(basename "$i")"
    nameNoExt="${name%.*}"
    outName="${out_biom_ITS}"/${nameNoExt}".biom"

    python "${qiimePathbiomformat}"/convert_biom.py \
         -i "$i" \
         -o "$outName" \
         --biom_table_type="otu table" \
         --process_obs_metadata taxonomy
done

##############
#            #
#    Oom     #
#            #
##############


for i in ${original_om}/*.tsv; do 
    name="$(basename "$i")"
    nameNoExt="${name%.*}"
    outName="${out_biom_Om}"/${nameNoExt}".biom"

    python "${qiimePathbiomformat}"/convert_biom.py \
         -i "$i" \
         -o "$outName" \
         --biom_table_type="otu table" \
         --process_obs_metadata taxonomy
done

##############
#            #
#    Phy     #
#            #
##############

for i in ${original_phy}/*.tsv; do 
    name="$(basename "$i")"
    nameNoExt="${name%.*}"
    outName="${out_biom_Phy}"/${nameNoExt}".biom"

    python "${qiimePathbiomformat}"/convert_biom.py \
         -i "$i" \
         -o "$outName" \
         --biom_table_type="otu table" \
         --process_obs_metadata taxonomy
done


######################################
#                                    #
#              Merge OTU             #
#                                    #
######################################

##############
#            #
#    ITS     #
#            #
##############

fileList=()
for i in $(find "$out_biom_ITS" -maxdepth 1 -type f -name "*biom"); do
    # echo "$i"
    fileList+=("$i")
done
#echo "${fileList[@]}"


allBiom=$(echo "${fileList[@]}" | tr " " ",")
#echo "$allBiom"

python "${qiimePath}"/merge_otu_tables.py \
         -i "$allBiom" \
         -o "${ITS_merged_biom}"/ITS_merged.biom


##############
#            #
#    Oom     #
#            #
##############

fileList=()
for i in $(find "$out_biom_Om" -maxdepth 1 -type f -name "*biom"); do
    # echo "$i"
    fileList+=("$i")
done
#echo "${fileList[@]}"


allBiom=$(echo "${fileList[@]}" | tr " " ",")
#echo "$allBiom"

python "${qiimePath}"/merge_otu_tables.py \
         -i "$allBiom" \
         -o "${Om_merged_biom}"/Om2_merged.biom


##############
#            #
#    Phy     #
#            #
##############

fileList=()
for i in $(find "$out_biom_Phy" -maxdepth 1 -type f -name "*biom"); do
    # echo "$i"
    fileList+=("$i")
done
#echo "${fileList[@]}"


allBiom=$(echo "${fileList[@]}" | tr " " ",")
#echo "$allBiom"

python "${qiimePath}"/merge_otu_tables.py \
         -i "$allBiom" \
         -o "${Phy_merged_biom}"/Phy_merged.biom


###################################################
#                                                 #
#       Convert biom tables in tsv tables         #
#                                                 #
###################################################

##############
#            #
#    ITS     #
#            #
##############

python "${qiimePathbiomformat}"/convert_biom.py \
      -i "${ITS_merged_biom}"/ITS_merged.biom \
      -o "$ITS_merged_tsv"/ITS_merged.tsv \
      -b \
      --header_key="taxonomy"

##############
#            #
#    Oom     #
#            #
##############

  python "${qiimePathbiomformat}"/convert_biom.py \
      -i "${Om_merged_biom}"/Om_merged.biom \
      -o "${Om_merged_tsv}"/Om_merged.tsv \
      -b \
      --header_key="taxonomy"

##############
#            #
#    Phy     #
#            #
##############

  python "${qiimePathbiomformat}"/convert_biom.py \
      -i "${Phy_merged_biom}"/Phy_merged.biom \
      -o "${Phy_merged_tsv}"/Phy_merged.tsv \
      -b \
      --header_key="taxonomy"

#end of cmd mode back to Rstudio
```
Create summary barplot and stats table about your OTU. Never use the taxFilled tables as it does not work.

```{r}
library("RAM")
# make new directory:
dir.create(paste(pathDataAn, "otuRecap/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture the path of the new directory:
otuDataPath <- paste(pathDataAn, "otuRecap/", sep = "")

# Create a new directory in otuRecap for plots:
dir.create(paste(otuDataPath, "Plots/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

# Capture path to new directory:
plotPath <- paste(otuDataPath, "Plots/", sep = "")

# Create a new directory in otuRecap for tables:
dir.create(paste(otuDataPath, "StatTables/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
statOTURecapPath <- paste(otuDataPath, "StatTables/", sep = "")



##############
#            #
#    ITS     #
#            #
############## WORKS 

  temp <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

# Open file-handle to get ready to make the png graph:
  png(filename=paste(plotPath, "ITS", ".otuRecap.png", sep = ""),
      width = 600, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Close the png graph file handle:
  dev.off()

df <- OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")
# Write out the dataframe:
  write.table(df, file=paste(statOTURecapPath, "ITS.otuRecap.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

##############
#            #
#    Oom     #
#            #
############## works

  temp <-  read.table(paste(pathDataAn, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

# Open file-handle to get ready to make the png graph:
  png(filename=paste(plotPath, "Om", ".otuRecap.png", sep = ""),
      width = 600, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Close the png graph file handle:
  dev.off()

df2 <-OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Write out the dataframe:
  write.table(df2, file=paste(statOTURecapPath, "Om.otuRecap.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
##############
#            #
#    Phy     #
#            #
############## works

temp <-  read.table(paste(pathDataAn, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

# Open file-handle to get ready to make the png graph:
  png(filename=paste(plotPath, "Phy", ".otuRecap.png", sep = ""),
      width = 600, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Close the png graph file handle:
  dev.off()

df3 <-OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Write out the dataframe:
  write.table(df3, file=paste(statOTURecapPath, "Phy.otuRecap.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
##############
#            #
#   Phy-nsp  #
#            #
############## works

temp <-  read.table(paste(pathDataAn, "Phy.nsp.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

# Open file-handle to get ready to make the png graph:
  png(filename=paste(plotPath, "Phy.nsp", ".otuRecap.png", sep = ""),
      width = 600, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Close the png graph file handle:
  dev.off()

df3 <-OTU.recap(list(data=temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greys")

# Write out the dataframe:
  write.table(df3, file=paste(statOTURecapPath, "Phy.nsp.otuRecap.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
```
tax.fill to remove uncertae sedis, unclassified...

```{r}
library("RAM")

taxFillDir <- "taxFillTables/"
dir.create(paste(pathDataAn, taxFillDir, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
taxFillPath <- paste(pathDataAn, taxFillDir, sep = "")


##############
#            #
#    ITS     #
#            #
############## works

temp <- read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))

  temp$row.names <- NULL
  temp <- tax.fill(temp, downstream=TRUE)
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", "ITS", ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)

##############
#            #
#    Oom     #
#            #
############## works
temp <- read.table(paste(pathDataAn, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))

  temp$row.names <- NULL
  temp <- tax.fill(temp, downstream=TRUE)
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", "Om", ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)
  
##############
#            #
#    Phy     #
#            #
############## works

temp <- read.table(paste(pathDataAn, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))

  temp$row.names <- NULL
  temp <- tax.fill(temp, downstream=TRUE)
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", "Phy", ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)

##############
#            #
#  Phy.nsp   #
#            #
############## works

temp <- read.table(paste(pathDataAn, "Phy.nsp.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))

  temp$row.names <- NULL
  temp <- tax.fill(temp, downstream=TRUE)
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", "Phy.nsp", ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)
  
  
```

Create diversity directory and path. List your datasets for diversity indexes. Adds a bunch of columns in a new diversity metadata file.
Indices obtained (in this specific order) are: Spec Number,  Simpson data,  Inv simpson data,	Shannon data,	Simpson eveness,	Shannon eveness,	Simpson true diversity,	shannon true diversity,	chao,	ACE.

```{r}
diversityDir <- paste(dataAnPath, "diversityDir/", sep ="")
dir.create(diversityDir,
           showWarnings = TRUE,
           recursive    = FALSE)

##############
#            #
#    ITS     #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "ITS.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

temp2 <- OTU.diversity(list(data=tabletemp), metatemp)

  write.table(temp2, file=paste(dataAnPath, "diversityDir/", "ITS", ".meta.div.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)

##############
#            #
#    Oom     #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "Om.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

temp2 <- OTU.diversity(list(data=tabletemp), metatemp)

  write.table(temp2, file=paste(dataAnPath, "diversityDir/", "Om", ".meta.div.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)

##############
#            #
#    Phy     #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "Phy.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL f not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL 

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

temp2 <- OTU.diversity(list(data=tabletemp), metatemp)

  write.table(temp2, file=paste(dataAnPath, "diversityDir/", "Phy", ".meta.div.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)


##############
#            #
# Phy.nsp    #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "Phy.nsp.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Phy.nsp.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL 

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

temp2 <- OTU.diversity(list(data=tabletemp), metatemp)

  write.table(temp2, file=paste(dataAnPath, "diversityDir/", "Phy.nsp", ".meta.div.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)  
  
```

Whiskers plots (diversity)
```{r}

diversityDir <- paste(dataAnPath, "diversityDir/EvennessPlot/", sep ="")
dir.create(diversityDir,
           showWarnings = TRUE,
           recursive    = FALSE)

diversityDir <- paste(dataAnPath, "diversityDir/DiversityPlot/", sep ="")
dir.create(diversityDir,
           showWarnings = TRUE,
           recursive    = FALSE)

# Capture the path of the new directory:
EvenPlotPath <- paste(dataAnPath, "diversityDir/EvennessPlot/", sep ="")
DivPlotPath <- paste(dataAnPath, "diversityDir/DiversityPlot/", sep ="")

##############
#            #
#    ITS     #
#            #
############## works


tabletemp <- read.table(paste(taxFillPath, "ITS.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(diversityPath, "ITS.meta.div.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work


#Simpson Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "ITS", ".Simpson.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "ITS", ".Shannon.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson True Diversity

# Open file-handle to get ready to make the png graph:

  png(filename=paste(DivPlotPath, "ITS", ".Simpson.True.Div.png", sep = ""),
      width = 1200, height = 600, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "ITS", ".Shannon.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", xlab = "Shannon True Diversity by Trap Type and their respective Lure",ylab="Units of number of species", facet.x.cex = "10", facet.y.cex = "10", facet.y = TRUE)

# Close the png graph file handle:
  dev.off()

#Simpson Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "ITS", ".Simpson.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "ITS", ".Shannon.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson inverted Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "ITS", ".Simpson.inv.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("invsim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson inv Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()


#Species number

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "ITS", ".Species.nb.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("spec"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Species nb", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

##############
#            #
#    Oom     #
#            #
############## works
  
tabletemp <- read.table(paste(taxFillPath, "Om.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(diversityPath, "Om.meta.div.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

#Simpson Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "Om", ".Simpson.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "Om", ".Shannon.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Om", ".Simpson.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Om", ".Shannon.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Om", ".Simpson.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Om", ".Shannon.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson inverted Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Om", ".Simpson.inv.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("invsim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson inv Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()


#Species number

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Om", ".Species.nb.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("spec"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Species nb", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

##############
#            #
#    Phy     #
#            #
############## works
  
tabletemp <- read.table(paste(taxFillPath, "Phy.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(diversityPath, "Phy.meta.div.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

#Simpson Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "Phy", ".Simpson.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "Phy", ".Shannon.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy", ".Simpson.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy", ".Shannon.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy", ".Simpson.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy", ".Shannon.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson inverted Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy", ".Simpson.inv.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("invsim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson inv Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()


#Species number

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy", ".Species.nb.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("spec"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Species nb", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

##############
#            #
#  Phy.nsp   #
#            #
############## works
  
tabletemp <- read.table(paste(taxFillPath, "Phy.nsp.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(diversityPath, "Phy.nsp.meta.div.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

#Simpson Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "Phy.nsp", ".Simpson.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Evenness

# Open file-handle to get ready to make the png graph:
  png(filename=paste(EvenPlotPath, "Phy.nsp", ".Shannon.Evenness.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_even"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Evenness", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy.nsp", ".Simpson.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon True Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy.nsp", ".Shannon.True.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan_trudiv"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon True Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy.nsp", ".Simpson.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("sim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Shannon Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy.nsp", ".Shannon.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("shan"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Shannon Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()

#Simpson inverted Diversity

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy.nsp", ".Simpson.inv.Div.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("invsim"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Simpson inv Div", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()


#Species number

# Open file-handle to get ready to make the png graph:
  png(filename=paste(DivPlotPath, "Phy.nsp", ".Species.nb.png", sep = ""),
      width = 1200, height = 400, units = "px")

myGroup.diversity(list(data=tabletemp), metatemp, factors = c("TrapType"), indices = c("spec"), diversity.info=TRUE, facet="TrapType", x.axis="Lure", compare="NULL", ylab="Species nb", facet.x.cex = "10", facet.y.cex = "10")

# Close the png graph file handle:
  dev.off()


```

Rarefy your tables
```{r}

dir.create(paste(pathDataAn, "rarefy_R/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
rarefyPath <- paste(pathDataAn, "rarefy_R/", sep = "")

##############
#            #
#    ITS     #
#            #
############## works

temp <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

rrf <- OTU.rarefy(list(data=temp), sample=NULL)
# Write out the dataframe:
  write.table(rrf, file=paste(rarefyPath, "ITS.rarefy.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

  
temp2 <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
  #temp$row.names <- NULL 

 


# Open file-handle to get ready to make the png graph:
  png(filename=paste(rarefyPath, "ITS", ".RAREFplot.png", sep = ""),
      width = 1200, height = 400, units = "px")

rrfPLOT <- rarecurve(temp2, step =10, xlab = "Sequence nb per sample", ylab = "Species OTUs") 

# Close the png graph file handle:
  dev.off()
  
##############
#            #
#    Oom     #
#            #
############## works

temp <-  read.table(paste(pathDataAn, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

rrf <- OTU.rarefy(list(data=temp), sample=NULL)
# Write out the dataframe:
  write.table(rrf, file=paste(rarefyPath, "Om.rarefy.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

##############
#            #
#    Phy     #
#            #
############## works

temp <-  read.table(paste(pathDataAn, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

rrf <- OTU.rarefy(list(data=temp), sample=NULL)
# Write out the dataframe:
  write.table(rrf, file=paste(rarefyPath, "Phy.rarefy.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

##############
#            #
#  Phy-nsp   #
#            #
############## works

temp <-  read.table(paste(pathDataAn, "Phy.nsp.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

rrf <- OTU.rarefy(list(data=temp), sample=NULL)
# Write out the dataframe:
  write.table(rrf, file=paste(rarefyPath, "Phy.nsp.rarefy.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

```

### group.abund.Taxa the bad one
#### top taxa at each rank (all ranks)
  + Needs more than one site in the OTU table to work properly.
  + top taxa at each rank (all ranks). In number or percent. Moustache plot.
  + Needs more than one site in the OTU table to work properly.
  + default top 10%
```{r}
library("RAM")
# make new directory:
dir.create(paste(pathDataAn, "groupTaxaBar/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
groupTaxaPath <- paste(pathDataAn, "groupTaxaBar/", sep = "")


###merged tables test


  tabletemp <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, #colClasses=c("row.names"="character", "taxonomy"="character"
                      ))
  tabletemp$row.names <- NULL 
  
   metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
metatemp$row.names <- NULL
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)]

     
# Open file-handle to get ready to make the png graph:
  png(filename=paste(groupTaxaPath, "ITS", ".groupTaxBar_kingdom.png", sep = ""),
      width = 1500, height = 1000, units = "px")

# Put this data into the png graph for the open filehandle:
#   group.Taxa.bar(data = list(data = trans), is.OTU = TRUE, func = "sum", rank="k", meta = temp, meta.factor = "TrapType", col.pal = "fullcolors")
group.Taxa.bar(data = list(data = tabletemp), is.OTU = TRUE, func = "sum", rank="k", meta = metatemp, meta.factor = "TrapType", col.pal = "Set3")

# Close the png graph file handle:
  dev.off()

}



```

##TaxAbund
### group.abund.Taxa
# 1. generate the list of taxa desired with core taxa, then generate the plots with group.abund.Taxa
```{r}
library("RAM")

# make new directory:
dir.create(paste(pathDataAn, "groupTaxaBar/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
groupTaxaPath <- paste(pathDataAn, "groupTaxaBar/", sep = "")


##############
#            #
#    ITS     #
#            #
##############

tabletemp <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

coreTaxaITSTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = TRUE, meta.factor = "TrapType", percent = 0.80)

taxaInsect <- coreTaxaITSTrapsp$data$Insect$taxa
taxaSoil   <- coreTaxaITSTrapsp$data$Soil$taxa
taxaSpore  <- c(unique(coreTaxaITSTrapsp$data$Spore$taxa, coreTaxaITSTrapsp$data$Spore_mini$taxa))
taxarotorod <- coreTaxaITSTrapsp$data$Rotorod$taxa
taxamock <- coreTaxaITSTrapsp$data$MOC$taxa
taxaAll    <- unique(c(taxaInsect, taxaSoil, taxaSpore, taxarotorod))
taxaAllbutsoil    <- unique(c(taxaInsect, taxaSpore, taxarotorod))

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.Insect.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaInsect,
                    main = "12 Species found in 80% of Insect samples and reprensenting 38.16% of total sequences in Insect Traps")
                
# Close the png graph file handle:
  dev.off()

coreTaxaITSTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "TrapType", percent = 0.80)
  
  # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.Insect.80pc.keep.unclassified.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaInsect,
                    main = "12 Species found in 80% of Insect samples and reprensenting 38.16% of total sequences in Insect Traps")
                
# Close the png graph file handle:
  dev.off()

    
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.soil.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaSoil,
                    main = "36 species found in 80% of Soil samples and representing 86.81% of total sequences in Soil samples")
                
# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.spore.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaSpore,
                    main = "15 species found in 80% of Spore samples and representing 61.47% of total sequences in Spore Traps")
                
# Close the png graph file handle:
  dev.off()
  
  # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.rotorod.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxarotorod,
                     main = "27 species found in 80% of Rotorod samples and representing 85.60% of total sequences in Rotorod samples")
                 
# Close the png graph file handle:
  dev.off()
 
  
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.mock.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxamock,
                     main = "9 species found in 80% of MOCK samples and representing 62.78% of total sequences in MOCK samples")
                
# Close the png graph file handle:
  dev.off()
      
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.all.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaAll,
                    main = "52 species found in 80% of samples in all Trap Types")
                
# Close the png graph file handle:
  dev.off()
  
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.allbutsoil.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaAllbutsoil,
                    main = "29 species found in 80% of samples in all Trap Types excluding soil samples")
                
# Close the png graph file handle:
  dev.off() 
  
   
  ##########
  #        #  
  #  Lures #
  #        #
  ##########
  
tabletemp <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work    
  
coreTaxaITSlureInssp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = TRUE, meta.factor = "Lure", percent = 0.80)

coreTaxaITSlureInssp
  
taxaAlphaPin <- coreTaxaITSlureInssp$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaITSlureInssp$data$C6C8$taxa
taxaEthylGly  <- coreTaxaITSlureInssp$data$Ethylene_glycol$taxa
taxaGenLongHorn <- coreTaxaITSlureInssp$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaITSlureInssp$data$Monochamus$taxa
taxaunknown <- coreTaxaITSlureInssp$data$unknown$taxa
taxaAll <- unique(c(taxaAlphaPin, taxaC6C8, taxaEthylGly, taxaGenLongHorn, taxaMonochamus, taxaunknown))

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.AlphaPin.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaAlphaPin,
                    main = "16 Species found in 80% of Insect samples with Alpha-Pinene Lure")
                
# Close the png graph file handle:
  dev.off()
  

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.C6C8.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaC6C8,
                    main = "20 Species found in 80% of Insect samples with C6C8")
                
# Close the png graph file handle:
  dev.off()  
    
 
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.EthylGly.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaEthylGly,
                    main = "1 Species found in 80% of Insect samples with Ehtylene-Glycol")
                
# Close the png graph file handle:
  dev.off()   
  
  
  # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.GenLongH.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaGenLongHorn,
                    main = "26 Species found in 80% of Insect samples with General Longhorn")
                
# Close the png graph file handle:
  dev.off()  

    # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.monochamus.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaMonochamus,
                    main = "18 Species found in 80% of Insect samples with Monochamus")
                
# Close the png graph file handle:
  dev.off()   
 
  
    # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "ITS", ".species.abund.unknown.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaunknown,
                    main = "56 Species found in 80% of Insect samples with unknown lure")
                
# Close the png graph file handle:
  dev.off()    
             
##############
#            #
#    Oom     #
#            #
############## works
  
tabletemp <-  read.table(paste(pathDataAn, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

coreTaxaOmTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = TRUE, meta.factor = "TrapType", percent = 0.80)

taxaInsect <- coreTaxaOmTrapsp$data$Insect$taxa
taxaSoil   <- coreTaxaOmTrapsp$data$Soil$taxa
taxaSpore  <- c(unique(coreTaxaOmTrapsp$data$Spore$taxa, coreTaxaOmTrapsp$data$Spore_mini$taxa))
taxarotorod <- coreTaxaOmTrapsp$data$Rotorod$taxa
taxamock <- coreTaxaOmTrapsp$data$MOC$taxa
taxaAll    <- unique(c(taxaInsect, taxaSoil, taxaSpore, taxarotorod))

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.Insect.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaInsect,
                    main = "12 Species found in 80% of Insect samples and reprensenting 38.16% of total sequences in Insect Traps")
                
# Close the png graph file handle:
  dev.off()

  
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.soil.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaSoil,
                    main = "36 species found in 80% of Soil samples and representing 86.81% of total sequences in Soil samples")
                
# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.spore.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaSpore,
                    main = "15 species found in 80% of Spore samples and representing 61.47% of total sequences in Spore Traps")
                
# Close the png graph file handle:
  dev.off()
  
  # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.rotorod.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxarotorod,
                     main = "27 species found in 80% of Rotorod samples and representing 85.60% of total sequences in Rotorod samples")
                 
# Close the png graph file handle:
  dev.off()
 
  
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.mock.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxamock,
                     main = "9 species found in 80% of MOCK samples and representing 62.78% of total sequences in MOCK samples")
                
# Close the png graph file handle:
  dev.off()
      
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.all.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaAll,
                    main = "7 species found in 80% of samples in all Trap Types and representing 48.76% of total sequences")
                
# Close the png graph file handle:
  dev.off()
           
   ##########
  #        #  
  #  Lures #
  #        #
  ##########
  
tabletemp <-  read.table(paste(pathDataAn, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work    
  
coreTaxaOmlureInssp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = TRUE, meta.factor = "Lure", percent = 0.80)

coreTaxaOmlureInssp
  
taxaAlphaPin <- coreTaxaOmlureInssp$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaOmlureInssp$data$C6C8$taxa
taxaGenLongHorn <- coreTaxaOmlureInssp$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaOmlureInssp$data$Monochamus$taxa
taxaunknown <- coreTaxaOmlureInssp$data$unknown$taxa

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.AlphaPin.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaAlphaPin,
                    main = "3 Species found in 80% of Insect samples with Alpha-Pinene Lure")
                
# Close the png graph file handle:
  dev.off()
  

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.C6C8.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaC6C8,
                    main = "18 Species found in 80% of Insect samples with C6C8")
                
# Close the png graph file handle:
  dev.off()  
    
 
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.monochamus.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaMonochamus,
                    main = "26 Species found in 80% of Insect samples with monochamus")
                
# Close the png graph file handle:
  dev.off()   
  
  
  # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.GenLongH.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaGenLongHorn,
                    main = "1 Species found in 80% of Insect samples with General Longhorn")
                
# Close the png graph file handle:
  dev.off()  


# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Om", ".species.abund.unknown.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaunknown,
                    main = "21 Species found in 80% of Insect samples with unknown lure")
                
# Close the png graph file handle:
  dev.off()    
  
  
##############
#            #
#    Phy     #
#            #
############## works  
  
tabletemp <-  read.table(paste(pathDataAn, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

coreTaxaPhyTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "TrapType", percent = 0.80)

taxaInsect <- coreTaxaPhyTrapsp$data$Insect$taxa
taxaSoil   <- coreTaxaPhyTrapsp$data$Soil$taxa
taxaSpore  <- c(unique(coreTaxaPhyTrapsp$data$Spore$taxa, coreTaxaPhyTrapsp$data$Spore_mini$taxa))
taxarotorod <- coreTaxaPhyTrapsp$data$Rotorod$taxa
taxamock <- coreTaxaPhyTrapsp$data$MOC$taxa
taxaAll    <- unique(c(taxaInsect, taxaSoil, taxaSpore, taxarotorod))

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.Insect.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaInsect,
                    main = "12 Species found in 80% of Insect samples and reprensenting 38.16% of total sequences in Insect Traps")
                
# Close the png graph file handle:
  dev.off()

  
# # Open file-handle to get ready to make the png graph:
#   png(filename=paste(taxabunDir, "Phy", ".species.abund.soil.80pc.png", sep = ""),
#       width = 1500, height = 1000, units = "px")
# 
# myGroup.abund.Taxa(data = list(data = tabletemp),
#                     is.OTU = TRUE,
#                     rank = "s",    # Pay attention to the rank you want to work on
#                     drop.unclassified = TRUE,
#                     meta = metatemp,
#                     meta.factor = "TrapType",
#                     taxa = taxaSoil,
#                     main = "36 species found in 80% of Soil samples and representing 86.81% of total sequences in Soil samples")
#                 
# # Close the png graph file handle:
#   dev.off()

# # Open file-handle to get ready to make the png graph:
#   png(filename=paste(taxabunDir, "Phy", ".species.abund.spore.80pc.png", sep = ""),
#       width = 1500, height = 1000, units = "px")
# 
# myGroup.abund.Taxa(data = list(data = tabletemp),
#                     is.OTU = TRUE,
#                     rank = "s",    # Pay attention to the rank you want to work on
#                     drop.unclassified = TRUE,
#                     meta = metatemp,
#                     meta.factor = "TrapType",
#                     taxa = taxaSpore,
#                     main = "15 species found in 80% of Spore samples and representing 61.47% of total sequences in Spore Traps")
#                 
# # Close the png graph file handle:
#   dev.off()
  
 # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.rotorod.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxarotorod,
                     main = "27 species found in 80% of Rotorod samples and representing 85.60% of total sequences in Rotorod samples")
                 
# Close the png graph file handle:
  dev.off()
 
  
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.mock.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxamock,
                     main = "9 species found in 80% of MOCK samples and representing 62.78% of total sequences in MOCK samples")
                
# Close the png graph file handle:
  dev.off()
      
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.all.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaAll,
                    main = "7 species found in 80% of samples in all Trap Types and representing 48.76% of total sequences")
                
# Close the png graph file handle:
  dev.off()
           
   ##########
  #        #  
  #  Lures #
  #        #
  ##########
  
tabletemp <-  read.table(paste(pathDataAn, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work    
  
coreTaxaPhylureInssp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = TRUE, meta.factor = "Lure", percent = 0.80)

coreTaxaPhylureInssp
  
taxaAlphaPin <- coreTaxaPhylureInssp$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaPhylureInssp$data$C6C8$taxa
taxaEthylGly  <- coreTaxaPhylureInssp$data$Ethylene_glycol$taxa
taxaGenLongHorn <- coreTaxaPhylureInssp$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaPhylureInssp$data$Monochamus$taxa
taxaunknown <- coreTaxaPhylureInssp$data$unknown$taxa

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.AlphaPin.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaAlphaPin,
                    main = "10 Species found in 80% of Insect samples with Alpha-Pinene Lure")
                
# Close the png graph file handle:
  dev.off()
  

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.C6C8.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaC6C8,
                    main = "3 Species found in 80% of Insect samples with C6C8")
                
# Close the png graph file handle:
  dev.off()  
    
 
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy", ".species.abund.monochamus.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaMonochamus,
                    main = "11 Species found in 80% of Insect samples with Monochamus")
                
# Close the png graph file handle:
  dev.off()   
 
  
##############
#            #
# Phy-nsp    #
#            #
############## works  
  
tabletemp <-  read.table(paste(pathDataAn, "Phy.nsp.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "Phy.nsp.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

coreTaxaPhynspTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "TrapType", percent = 0.80)

taxaInsect <- coreTaxaPhynspTrapsp$data$Insect$taxa
taxaSoil   <- coreTaxaPhynspTrapsp$data$Soil$taxa
taxaSpore  <- c(unique(coreTaxaPhynspTrapsp$data$Spore$taxa, coreTaxaPhynspTrapsp$data$Spore_mini$taxa))
taxarotorod <- coreTaxaPhynspTrapsp$data$Rotorod$taxa
taxamock <- coreTaxaPhynspTrapsp$data$MOC$taxa
taxaAll    <- unique(c(taxaInsect, taxaSoil, taxaSpore, taxarotorod))

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.Insect.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaInsect,
                    main = "Species found in 80% of Insect samples")
                
# Close the png graph file handle:
  dev.off()

  
# Open file-handle to get ready to make the png graph:
png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.soil.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")
myGroup.abund.Taxa(data = list(data = tabletemp),
                  is.OTU = TRUE,
                   rank = "s",    # Pay attention to the rank you want to work on
                  drop.unclassified = TRUE,
                  meta = metatemp,
                  meta.factor = "TrapType",
                  taxa = taxaSoil,
                  main = "species found in 80% of Soil samples")

# # Close the png graph file handle:
dev.off()

# Open file-handle to get ready to make the png graph:
png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.spore.80pc.png", sep = ""),
 width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                  is.OTU = TRUE,
                   rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                   meta = metatemp,
                   meta.factor = "TrapType",
                   taxa = taxaSpore,
                    main = "species found in 80% of Spore samples")

# # Close the png graph file handle:
dev.off()
  
 # Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.rotorod.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxarotorod,
                     main = "species found in 80% of Rotorod samples")
                 
# Close the png graph file handle:
  dev.off()
 
  
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.mock.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxamock,
                     main = "species found in 80% of MOCK samples")
                
# Close the png graph file handle:
  dev.off()
      
# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.all.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    # Pay attention to the rank you want to work on
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "TrapType",
                    taxa = taxaAll,
                    main = "species found in 80% of samples")
                
# Close the png graph file handle:
  dev.off()
           
   ##########
  #        #  
  #  Lures #
  #        #
  ##########
  
tabletemp <-  read.table(paste(pathDataAn, "Phy.nsp.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
tabletemp$row.names <- NULL 
  
metatemp <-  read.table(paste(pathDataAn, "Phy.nsp.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work    
  
coreTaxaPhynsplureInssp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = TRUE, meta.factor = "Lure", percent = 0.80)

coreTaxaPhylureInssp
  
taxaAlphaPin <- coreTaxaPhynsplureInssp$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaPhynsplureInssp$data$C6C8$taxa
taxaEthylGly  <- coreTaxaPhynsplureInssp$data$Ethylene_glycol$taxa
taxaGenLongHorn <- coreTaxaPhynsplureInssp$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaPhynsplureInssp$data$Monochamus$taxa
taxaunknown <- coreTaxaPhynsplureInssp$data$unknown$taxa

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.AlphaPin.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaAlphaPin,
                    main = "Species found in 80% of Insect samples")
                
# Close the png graph file handle:
  dev.off()
  

# Open file-handle to get ready to make the png graph:
  png(filename=paste(taxabunDir, "Phy.nsp", ".species.abund.C6C8.80pc.png", sep = ""),
      width = 1500, height = 1000, units = "px")

myGroup.abund.Taxa(data = list(data = tabletemp),
                    is.OTU = TRUE,
                    rank = "s",    
                    drop.unclassified = TRUE,
                    meta = metatemp,
                    meta.factor = "Lure",
                    taxa = taxaC6C8,
                    main = "Species found in 80% of Insect samples with C6C8")
                
# Close the png graph file handle:
  dev.off()  
    
 
# # Open file-handle to get ready to make the png graph:
#   png(filename=paste(taxabunDir, "Ph.nsp", ".species.abund.monochamus.80pc.png", sep = ""),
#       width = 1500, height = 1000, units = "px")
# 
# myGroup.abund.Taxa(data = list(data = tabletemp),
#                     is.OTU = TRUE,
#                     rank = "s",    
#                     drop.unclassified = TRUE,
#                     meta = metatemp,
#                     meta.factor = "Lure",
#                     taxa = taxaMonochamus,
#                     main = "Species found in 80% of Insect samples with Monochamus")
#                 
# # Close the png graph file handle:
#   dev.off()   
 
  
```


# Produce the rarefaction curve for each sample (copied from Wen)

# First transpose the otu table, because the rarecurve take each row as one sample

```{r}
TransposedOTUTables <- "transposed_OTU_tables/"
dir.create(paste(dataAnPath, TransposedOTUTables, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
TransposedOTUTables <- paste(dataAnPath, TransposedOTUTables, sep = "")

##############
#            #
#    ITS     #
#            #
############## works

tmp <-  read.table(paste(taxFillPath, "ITS.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tmp2 <- transpose.OTU(tmp)
write.table(tmp2, file=paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)



##############
#            #
#    Oom     #
#            #
############## works

tmp <-  read.table(paste(taxFillPath, "Om.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tmp2 <- transpose.OTU(tmp)
write.table(tmp2, file=paste(TransposedOTUTables, "Om.trans.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
##############
#            #
#    Phy     #
#            #
############## works

tmp <-  read.table(paste(taxFillPath, "Phy.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tmp2 <- transpose.OTU(tmp)
write.table(tmp2, file=paste(TransposedOTUTables, "Phy.trans.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#########nope

#do a barplot of frequency of abundance ranks

# Open file-handle to get ready to make the barplot:
  png(filename=paste(rarefyPath, "ITS", ".freq_barplot.png", sep = ""),
      width = 1200, height = 1000)

barplot(log(abITS), las=1, xlab = "Abundance in Matrix", ylab = "log (frequency)", ylim = c(0, max(log(abITS))+2))

# Close the png graph file handle:
  dev.off() 
  
  ###

abITS
x11(width = 12, height = 4)
# do a barplot of frequency of abundance ranks
barplot(log(tmp), las=1, xlab = "Abundand in matrix", ylab = "log (frequency)", ylim = c(0, max(log(tmp))+2))

  
######
tmp <-  read.table(paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
  
  
abITS<-table(unlist(tmp))
abOm<-table(unlist(paste(TransposedOTUTables, "Om.trans.table.tsv", sep = "")))
abPhy<-table(unlist(paste(TransposedOTUTables, "Phy.trans.table.tsv", sep = "")))

# do a barplot of frequency of abundance ranks
barplot(log(ab), las=1, xlab = "Abundand in matrix", ylab = "log (frequency)", ylim = c(0, max(log(ab))+2))

# Total count/proportion of absences
sum(ITS1.t == 0)
sum(ITS2.t == 0)
sum(ITS1.t == 0)/(nrow(ITS1.t)*ncol(ITS1.t))
sum(ITS2.t == 0)/(nrow(ITS2.t)*ncol(ITS2.t))



abITS
X11(width = 12, height = 4)

# do a barplot of frequency of abundance ranks
ITSbplotAB <-barplot(log(abITS), las=1, xlab = "Abundand in matrix", ylab = "log (frequency)", ylim = c(0, max(log(abITS))+2))

# Total count/proportion of absences
sum((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")) == 0)
sum((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")) == 0)/(nrow((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")))*ncol((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = ""))))


```

# plot rarecurve for each sample

```{r}

#does not work
library("vegan")
#tiff(compression="lzw", file=paste(basename, sep="", "ITS.rarecurve.tiff"), res=1000, units="in", width=16, height=8)

x11(width=8, height=4)
par(mfrow=c(1,2))
rarecurve((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")),step=20, sample=min(rowSums((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")))), col="blue", cex=0.6, xlab="Number of sequences", ylab="Number of OTUs", main="ITS")

dev.off()
##

tmp <-  read.table(paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


# Open file-handle to get ready to make the barplot:
  png(filename=paste(rarefyPath, "ITS", ".rarecurve.png", sep = ""), res ="1000", units = "px",
      width = 1200, height = 1000)
rarecurve((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")),step=20, sample=min(rowSums((paste(TransposedOTUTables, "ITS.trans.table.tsv", sep = "")))), col="blue", cex=0.6, xlab="Number of sequences", ylab="Number of OTUs", main="ITSFungi")
#barplot(log(abITS), las=1, xlab = "Abundance in Matrix", ylab = "log (frequency)", ylim = c(0, max(log(abITS))+2))

# Close the png graph file handle:
  dev.off() 
  
##
#calculate species accumulation curves for subsets of the community and environmental data sets
library(BiodiversityR)

x11(width=8, height=4)
par(mfrow=c(1,2))
accumcomp(ITS1.t, y=meta, factor='Crop', method="rarefaction",cex=0.5, ylab="OTU number", xlab="Samples") # need to place the lengend by click on the plot before moving to next plot
accumcomp(ITS1.t, y=meta, factor='City', method="rarefaction",cex=0.5, ylab="OTU number", xlab="Samples")
dev.off()

#dev.copy(png, file="ITS1.otu.rarecurve.eachsample.png")

#dev.off()




# Open file-handle to get ready to make the raref curve:
  png(filename=paste(dataAnPath/trans, metadataMasterTax$ShortName[i], ".otuRecap.png", sep = ""),
      width = 800, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(data = list(data = temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Set3")

# Close the png graph file handle:
  dev.off()


```{r}
ITS1.t<-t(ITS1[, -dim(ITS1)[2]])
head(ITS1.t)[,1:5]
ITS2.t<-t(ITS2[, -dim(ITS2)[2]])
head(ITS2.t)[,1:5]

##from forum##
library(reshape2)
dcast.data.table(melt(mydata, id.vars = "col0"), variable ~ col0)
###
ab<-table(unlist(ITS1.t))
ab
x11(width = 12, height = 4)
# do a barplot of frequency of abundance ranks
barplot(log(ab), las=1, xlab = "Abundand in matrix", ylab = "log (frequency)", ylim = c(0, max(log(ab))+2))

# Total count/proportion of absences
sum(ITS1.t == 0)
sum(ITS2.t == 0)
sum(ITS1.t == 0)/(nrow(ITS1.t)*ncol(ITS1.t))
sum(ITS2.t == 0)/(nrow(ITS2.t)*ncol(ITS2.t))

```

# plot rarecurve for each sample DOES NT WORK

```{r}
library("vegan")
#tiff(compression="lzw", file=paste(basename, sep="", "ITS.rarecurve.tiff"), res=1000, units="in", width=16, height=8)

x11(width=8, height=4)
par(mfrow=c(1,2))
rarecurve(ITS1.t,step=20, sample=min(rowSums(ITS1.t)), col="blue", cex=0.6, xlab="Number of sequences", ylab="Number of OTUs", main="ITS1")
rarecurve(ITS2.t,step=20, sample=min(rowSums(ITS2.t)), col="blue", cex=0.6, xlab="Number of sequences", ylab="Number of OTUs", main="ITS2")
dev.off()

#calculate species accumulation curves for subsets of the community and environmental data sets
library(BiodiversityR)

x11(width=8, height=4)
par(mfrow=c(1,2))
accumcomp(ITS1.t, y=meta, factor='Crop', method="rarefaction",cex=0.5, ylab="OTU number", xlab="Samples") # need to place the lengend by click on the plot before moving to next plot
accumcomp(ITS1.t, y=meta, factor='City', method="rarefaction",cex=0.5, ylab="OTU number", xlab="Samples")
dev.off()

#dev.copy(png, file="ITS1.otu.rarecurve.eachsample.png")

#dev.off()
```

# GroupTaxabar

```{r}

#dir.create(paste(dataAnPath, "Ranks", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

#RanksPath <- paste(dataAnPath, "Ranks", sep = "")

tabletemp <-  read.table(paste(taxFillPath, "ITS.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

##############
#            #
#    ITS     #
#            #
############## works


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.sp.lure.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the species level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()

tmp <- group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 5, count =  FALSE, drop.unclassified = TRUE, meta = metatemp, meta.factor = c("Lure"))  

   
# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.ge.lure.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="g", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the genus level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.ki.lure.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="k", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the kingdom level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.ph.lure.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)


group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()
##############

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.sp.trap.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the species level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.ge.trap.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="g", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the genus level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.ki.trap.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="k", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the kingdom level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "ITS", ".gr.abund.ph.trap.png", sep = ""),
      width = 30000, height = 800, units = "px", pointsize = 10)


group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

tmp <-   group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Trap", meta = metatemp, meta.factor = c("TrapType"))
  
  
##############
#            #
#    Oom     #
#            #
############## works

tabletemp <-  read.table(paste(taxFillPath, "Om.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work



# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.sp.lure.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the species level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.ge.lure.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="g", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the genus level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.ki.lure.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="k", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the kingdom level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.ph.lure.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)


group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()
##############

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.sp.trap.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the species level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.ge.trap.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="g", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the genus level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.ki.trap.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="k", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the kingdom level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Om", ".gr.abund.ph.trap.png", sep = ""),
      width = 8000, height = 800, units = "px", pointsize = 10)


group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

##############
#            #
#    Phy     #
#            #
############## works

tabletemp <-  read.table(paste(taxFillPath, "Phy.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work



# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.sp.lure.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the species level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.ge.lure.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="g", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the genus level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.ki.lure.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="k", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the kingdom level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.ph.lure.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)


group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Lure", meta = metatemp, meta.factor = c("Lure"))

# Close the png graph file handle:
  dev.off()
##############

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.sp.trap.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="s", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the species level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.ge.trap.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="g", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the genus level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.ki.trap.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)

group.abundance.meta(data=(list(data=tabletemp)), rank="k", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the kingdom level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()


# Open file-handle to get ready to make the plot:
  png(filename=paste(groupTaxaPath, "Phy", ".gr.abund.ph.trap.png", sep = ""),
      width = 1000, height = 800, units = "px", pointsize = 10)


group.abundance.meta(data=(list(data=tabletemp)), rank="p", top = 25, count =  FALSE, drop.unclassified = TRUE, cex.x = 10, main = "OTU abundance at the phylum level by Trap", meta = metatemp, meta.factor = c("TrapType"))

# Close the png graph file handle:
  dev.off()

```
# GPS stuff
#you must remove samples with no gsp coordinates, example: mocks

```{r}


# make new directory:
dir.create(paste(dataAnPath, "geographics/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture the path of the new directory:
geoDir <- paste(pathDataAn, "geographics/", sep = "")



##############
#            #
#    ITS     #
#            #
############## works with no Na in table

metatemp <-  read.table(paste(dataAnPath, "ITS.meta.nomock.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL


# Open file-handle to get ready to make the png graph:
  png(filename=paste(geoDir, "ITS", ".geo.png", sep = ""),
      width = 1200, height = 1000, units = "px")

sample.sites(meta = metatemp, zoom=21, lat = "Latitude", lon = "Longitude", maptype="terrain")

# Close the png graph file handle:
  dev.off()
  
##############
#            #
#    Oom     #
#            #
############## works

metatemp <-  read.table(paste(dataAnPath, "Om.meta.nomock.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL


# Open file-handle to get ready to make the png graph:
  png(filename=paste(geoDir, "Om", ".geo.png", sep = ""),
      width = 1200, height = 1000, units = "px")

sample.sites(meta = metatemp, zoom=21, lat = "Latitude", lon = "Longitude", maptype="terrain")

# Close the png graph file handle:
  dev.off()
  
##############
#            #
#    Phy     #
#            #
############## works

metatemp <-  read.table(paste(dataAnPath, "Phy.meta.nomock.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL


# Open file-handle to get ready to make the png graph:
  png(filename=paste(geoDir, "Phy", ".geo.png", sep = ""),
      width = 1200, height = 1000, units = "px")

sample.sites(meta = metatemp, zoom=21, lat = "Latitude", lon = "Longitude", maptype="terrain")

# Close the png graph file handle:
  dev.off() 
 
```  
  
PCOA  
  
```{r}

pcoaDir <- paste(dataAnPath, "PCOA/", sep ="")
dir.create(pcoaDir,
           showWarnings = TRUE,
           recursive    = FALSE)

##############
#            #
#    ITS     #
#            #
##############

tabletemp <- read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, row.names = 1)
tabletemp$row.names <- NULL #if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

myvec <- c("TrapType")

names(myvec) <- "TrapType"

#either euclidean and normalize or hellinger and bray or jaccard (pick the one that represent more % of your data)
#stand.method	One of "total", "max", "freq", "normalize", "range", "standardize", "pa", "chi.square", "hellinger" or "log"
#dist.method	one of "manhattan", "euclidean", "canberra", "bray", "kulczynski", "jaccard", "gower", "altGower","morisita", "horn", "mountford", "raup", "binomial", "chao", or "cao"

# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "ITS", ".PCOA.disteuc.standnorm.TrapType.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = (myvec), rank = NULL, stand.method = "normalize", dist.method = "euclidean", sample.labels = FALSE, top = 20, ellipse = FALSE, main = "PCOA by Trap Type for Fungi", ggplot2 = TRUE, bw = TRUE)

# Close the png graph file handle:
  dev.off() 
 
  # Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "ITS", ".PCOA.distHell.standBray.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = (myvec), rank = NULL, stand.method = "hellinger", dist.method = "bray", sample.labels = FALSE, top = 5, ellipse = FALSE, main = "PCOA by Trap Type for Fungi", ggplot2 = FALSE, bw = FALSE)

# Close the png graph file handle:
  dev.off() 
  

    # Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "ITS", ".PCOA.distHell.standJaccard.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = (myvec), rank = NULL, stand.method = "hellinger", dist.method = "jaccard", sample.labels = FALSE, top = 5, ellipse = FALSE, main = "PCOA by Trap Type for Fungi", ggplot2 = FALSE, bw = FALSE)

# Close the png graph file handle:
  dev.off() 
  
  
# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "ITS", ".PCOA.disteuc.standnorm.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = "TrapType", rank = NULL, stand.method = "normalize", dist.method = "euclidean", sample.labels = FALSE, top = 5, ellipse = FALSE, main = "PCOA by Trap Type for Fungi", ggplot2 = FALSE, bw = FALSE)

# Close the png graph file handle:
  dev.off()   
     
  
# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "ITS", ".PCOA.disteuc.standnormBW.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = c("TrapType"), rank = NULL, stand.method = "normalize", dist.method = "euclidean", sample.labels = FALSE, top = 5, ellipse = 1, main = "PCOA by Trap Type for Fungi", ggplot2 = FALSE, bw = TRUE)

# Close the png graph file handle:
  dev.off()    
  
  
# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "ITS", ".PCOA.disteuc.standnormELlipse.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = c("TrapType"), rank = NULL, stand.method = "normalize", dist.method = "euclidean", sample.labels = FALSE, top = 5, ellipse = 1, main = "PCOA by Trap Type for Fungi", ggplot2 = FALSE, bw = FALSE)

# Close the png graph file handle:
  dev.off()   
  
##############
#            #
#    Oom     #
#            #
############## works
  
tabletemp <- read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, row.names = 1)
tabletemp$row.names <- NULL #if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, row.names =1)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

myvec <- c("TrapType")

names(myvec) <- "TrapType"

 # Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "Om", "PCOA.disteuc.standNorm.TrapType.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = myvec, rank = NULL, stand.method = "normalize", dist.method = "euclidean", sample.labels = FALSE, top = 20, ellipse = 1, main = "PCOA by Trap Type for Oomycetes", ggplot2 = TRUE, bw = FALSE)

# Close the png graph file handle:
  dev.off()

# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "Om", "PCOA.distbray.standHell.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = "TrapType", rank = NULL, stand.method = "hellinger", dist.method = "bray", sample.labels = FALSE, top = 20, ellipse = FALSE, main = "PCOA by Trap Type for Oomycetes", ggplot2 = TRUE, bw = FALSE)

# Close the png graph file handle:
  dev.off() 
 
  
   
  # Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "Om", "PCOA.distJaccard.standHell.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = "TrapType", rank = NULL, stand.method = "hellinger", dist.method = "jaccard", sample.labels = FALSE, top = 20, ellipse = FALSE, main = "PCOA by Trap Type for Oomycetes", ggplot2 = TRUE, bw = FALSE)

# Close the png graph file handle:
  dev.off() 
  
```  
```{r}

##############
#            #
#    Phy     #
#            #
############## 

tabletemp <- read.table(paste(dataAnPath, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, row.names = 1)
tabletemp$row.names <- NULL #if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

myvec <- c(TrapType = "TrapType")
names(myvec) <- "TrapType"

# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "Phy", "PCOA.distbray.standHell.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = (myvec), rank = NULL, stand.method = "hellinger", dist.method = "bray", sample.labels = FALSE, top = 5, ellipse = FALSE, main = "PCOA by Trap Type for Phytophthora", ggplot2 = FALSE, bw = FALSE)

# Close the png graph file handle:
  dev.off() 
  
  
# Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "Phy", "PCOA.distJaccard.standHell.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = (myvec), rank = NULL, stand.method = "hellinger", dist.method = "jaccard", sample.labels = FALSE, top = 20, ellipse = FALSE, main = "PCOA by Trap Type for Phytophthora", ggplot2 = FALSE, bw = FALSE)

# Close the png graph file handle:
  dev.off()   
  
 # Open file-handle to get ready to make the png graph:
png(filename=paste(pcoaDir, "Phy", "PCOA.distEuc.standNorm.TrapType.png", sep = ""),
      width = 1200, height = 1000, units = "px")

pcoa.plot(data = tabletemp, is.OTU = TRUE, meta = metatemp, factors = (myvec), rank = NULL, stand.method = "normalize", dist.method = "euclidean", sample.labels = FALSE, top = 20, ellipse = FALSE, main = "PCOA by Trap Type for Phytophthora", ggplot2 = TRUE, bw = TRUE)

# Close the png graph file handle:
  dev.off()    
  
  
```

#Tax abund tables generated at each taxonomic rank
```{r}

taxabunDir <- paste(dataAnPath, "taxAbund/", sep ="")
dir.create(taxabunDir,
           showWarnings = TRUE,
           recursive    = FALSE)

##############
#            #
#    ITS     #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "ITS.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

tmp <- tax.abund(tabletemp, otu2 = NULL, rank = NULL, drop.unclassified =TRUE, top = NULL, mode = "percent")
write.table(tmp, file=paste(taxabunDir, "ITS.taxab.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
                               
##############
#            #
#    Oom     #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "Om.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

tmp <- tax.abund(tabletemp, otu2 = NULL, rank = NULL, drop.unclassified =TRUE, top = NULL, mode = "percent")
write.table(tmp, file=paste(taxabunDir, "Om.taxab.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)      
                               
                               
##############
#            #
#    Phy     #
#            #
############## works

tabletemp <- read.table(paste(taxFillPath, "Phy.table.taxFill.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
#tabletemp$row.names <- NULL if not using tXfilled tables you need it
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)

#metatemp$row.names <- NULL

rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work

tmp <- tax.abund(tabletemp, otu2 = NULL, rank = NULL, drop.unclassified =TRUE, top = NULL, mode = "percent")
write.table(tmp, file=paste(taxabunDir, "Phy.taxab.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)           

```
rarecurve sort of working lol

```{r}

normalize your data through qiime

normDir <- paste(dataAnPath, "norm/", sep ="")
dir.create(normDir,
           showWarnings = TRUE,
           recursive    = FALSE)

#in terminal#
activate qiime

#"activate" qiime 
source /isilon/biodiversity/pipelines/qiime-1.7.0/activate.sh

dataAnPath <- "/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/"
normDir <- "/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/norm/"



###################################################
#                                                 #
#       Convert .tsv tables in biom tables        #
#                                                 #
###################################################

##############
#            #
#    ITS     #
#            #
##############


for i in ${dataAnPath}/*.table.tsv; do 
    name="$(basename "$i")"
    nameNoExt="${name%.*}"
    outName="${normDir}"/${nameNoExt}".biom"

    python "${qiimePathbiomformat}"/convert_biom.py \
         -i "$i" \
         -o "$outName" \
         --biom_table_type="otu table" \
         --process_obs_metadata taxonomy
done

##############
#            #
#    Oom     #
#            #
##############


for i in ${original_om}/*.tsv; do 
    name="$(basename "$i")"
    nameNoExt="${name%.*}"
    outName="${out_biom_Om}"/${nameNoExt}".biom"

    python "${qiimePathbiomformat}"/convert_biom.py \
         -i "$i" \
         -o "$outName" \
         --biom_table_type="otu table" \
         --process_obs_metadata taxonomy
done

##############
#            #
#    Phy     #
#            #
##############

for i in ${original_phy}/*.tsv; do 
    name="$(basename "$i")"
    nameNoExt="${name%.*}"
    outName="${out_biom_Phy}"/${nameNoExt}".biom"

    python "${qiimePathbiomformat}"/convert_biom.py \
         -i "$i" \
         -o "$outName" \
         --biom_table_type="otu table" \
         --process_obs_metadata taxonomy
done


######################################
#                                    #
#              Merge OTU             #
#                                    #
######################################

##############
#            #
#    ITS     #
#            #
##############

fileList=()
for i in $(find "$out_biom_ITS" -maxdepth 1 -type f -name "*biom"); do
    # echo "$i"
    fileList+=("$i")
done
#echo "${fileList[@]}"


allBiom=$(echo "${fileList[@]}" | tr " " ",")
#echo "$allBiom"

python "${qiimePath}"/merge_otu_tables.py \
         -i "$allBiom" \
         -o "${ITS_merged_biom}"/ITS_merged.biom


##############
#            #
#    Oom     #
#            #
##############

fileList=()
for i in $(find "$out_biom_Om" -maxdepth 1 -type f -name "*biom"); do
    # echo "$i"
    fileList+=("$i")
done
#echo "${fileList[@]}"


allBiom=$(echo "${fileList[@]}" | tr " " ",")
#echo "$allBiom"

python "${qiimePath}"/merge_otu_tables.py \
         -i "$allBiom" \
         -o "${Om_merged_biom}"/Om2_merged.biom


##############
#            #
#    Phy     #
#            #
##############

fileList=()
for i in $(find "$out_biom_Phy" -maxdepth 1 -type f -name "*biom"); do
    # echo "$i"
    fileList+=("$i")
done
#echo "${fileList[@]}"


allBiom=$(echo "${fileList[@]}" | tr " " ",")
#echo "$allBiom"

python "${qiimePath}"/merge_otu_tables.py \
         -i "$allBiom" \
         -o "${Phy_merged_biom}"/Phy_merged.biom


###################################################
#                                                 #
#       Convert biom tables in tsv tables         #
#                                                 #
###################################################

##############
#            #
#    ITS     #
#            #
##############

python "${qiimePathbiomformat}"/convert_biom.py \
      -i "${ITS_merged_biom}"/ITS_merged.biom \
      -o "$ITS_merged_tsv"/ITS_merged.tsv \
      -b \
      --header_key="taxonomy"

##############
#            #
#    Oom     #
#            #
##############

  python "${qiimePathbiomformat}"/convert_biom.py \
      -i "${Om_merged_biom}"/Om_merged.biom \
      -o "${Om_merged_tsv}"/Om_merged.tsv \
      -b \
      --header_key="taxonomy"

##############
#            #
#    Phy     #
#            #
##############

  python "${qiimePathbiomformat}"/convert_biom.py \
      -i "${Phy_merged_biom}"/Phy_merged.biom \
      -o "${Phy_merged_tsv}"/Phy_merged.tsv \
      -b \
      --header_key="taxonomy"

#end of cmd mode back to Rstudio


##############
#            #
#    ITS     #
#            #
##############

temp <-  read.table(paste(pathDataAn, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
temp$taxonomy <- NULL 
temp$row.names <- NULL  

tempSPEC <-  read.table(paste(diversityPath, "ITS.meta.div.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)



# Open file-handle to get ready to make the png graph:
png(filename=paste(rarefyPath, "ITS", "rarecurve2.png", sep = ""),
      width = 1200, height = 1000, units = "px")
rarecurve(temp, step = 250000, xlab = "Sample Size", ylab = "Species", label = TRUE)  
  
# Close the png graph file handle:
  dev.off() 
  
  
  # Open file-handle to get ready to make the png graph:
png(filename=paste(rarefyPath, "ITS", "rarecurve3.png", sep = ""),
      width = 1200, height = 1000, units = "px")
rarecurve(temp, step = 10, xlab = "Sample Size", ylab = "Species", label = TRUE)  
  
# Close the png graph file handle:
  dev.off() 
  
  
  # Open file-handle to get ready to make the png graph:
png(filename=paste(rarefyPath, "ITS", "rarecurve4.png", sep = ""),
      width = 1200, height = 1000, units = "px")
rarecurve(temp, step = 1, xlab = "Sample Size", ylab = "Species", label = TRUE)  
  
# Close the png graph file handle:
  dev.off()   

##############
#            #
#    Oom     #
#            #
############## works


temp <-  read.table(paste(pathDataAn, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
temp$taxonomy <- NULL 
temp$row.names <- NULL  

# Open file-handle to get ready to make the png graph:
png(filename=paste(rarefyPath, "Om", "rarecurve.png", sep = ""),
      width = 1200, height = 1000, units = "px")
rarecurve(temp, step = 1, xlab = "Sample Size", ylab = "Species", label = TRUE)  
  
# Close the png graph file handle:
  dev.off() 
    
  
  
```
Word search in Perl find species and sequence and other meta info
```{r}

#########################
#                       #
#    ITS by species     #
#                       #
#########################

#Define your 6 arguments to run the perl script
query_in <- paste(WordMatchDir, "speciesQuery.txt", sep = "")
otu_in <- paste(pathDataAn, "ITS.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "ITS.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_ITS.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "ITS_species_query_by_taxon.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "ITS_species_query_by_sample.tsv", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractor.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             sep = " ")
system(cmd)

#######################
#                     #
#    ITS by genus     #
#                     #
#######################

query_in <- paste(WordMatchDir, "genusQuery.txt", sep = "")
otu_in <- paste(pathDataAn, "ITS.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "ITS.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_ITS.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "ITS_genus_query_by_taxon.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "ITS_genus_query_by_sample.tsv", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractor.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             sep = " ")
system(cmd)


#######################
#                     #
#    Om by genus      #
#                     #
#######################

query_in <- paste(WordMatchDir, "genusOm.txt", sep = "")
otu_in <- paste(pathDataAn, "Om.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "Om.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_Om.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "Om_genus_query_by_taxon.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "Om_genus_query_by_sample.tsv", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractor.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             sep = " ")
system(cmd)


#######################
#                     #
#    Phy by genus     #
#                     #
#######################

query_in <- paste(WordMatchDir, "genusOm.txt", sep = "")
otu_in <- paste(pathDataAn, "Phy.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "Phy.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_Phy.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "Phy_genus_query_by_taxon.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "Phy_genus_query_by_sample.tsv", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractor.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             sep = " ")
system(cmd)

```
#Worsearch and FASTA improved
```{r}

#########################
#                       #
#    ITS by species     #
#                       #
#########################

#Define your 6 arguments to run the perl script
query_in <- paste(WordMatchDir, "speciesQuery.txt", sep = "")
otu_in <- paste(pathDataAn, "ITS.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "ITS.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_ITS.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "ITS_species_query_by_taxon2.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "ITS_species_query_by_sample2.tsv", sep = "")
fasta_out_folder <- paste(WordMatchDir, "/FastaOut/ITS_species_query", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractorGOOD.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             fasta_out_folder,
             sep = " ")
system(cmd)



#########################
#                       #
#    ITS by genuses     #
#                       #
#########################

#Define your 6 arguments to run the perl script
query_in <- paste(WordMatchDir, "genusQuery.txt", sep = "")
otu_in <- paste(pathDataAn, "ITS.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "ITS.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_ITS.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "ITS_genus_query_by_taxon2.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "ITS_genus_query_by_sample2.tsv", sep = "")
fasta_out_folder <- paste(WordMatchDir, "/FastaOut/ITS_genus_query", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractorGOOD.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             fasta_out_folder,
             sep = " ")
system(cmd)


###############
#             #
#    Oom      #
#             #
###############

#Define your 6 arguments to run the perl script
query_in <- paste(WordMatchDir, "genusOm.txt", sep = "")
otu_in <- paste(pathDataAn, "Om.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "Om.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_Om.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "Om_query_by_taxon2.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "Om_query_by_sample2.tsv", sep = "")
fasta_out_folder <- paste(WordMatchDir, "/FastaOut/om_query", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractorGOOD.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             fasta_out_folder,
             sep = " ")
system(cmd)



##############
#            #
#    PHY     #
#            #
##############

#Define your 6 arguments to run the perl script
query_in <- paste(WordMatchDir, "genusOm.txt", sep = "")
otu_in <- paste(pathDataAn, "Phy.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "Phy.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_Phy.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "Phy_query_by_taxon2.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "Phy_query_by_sample2.tsv", sep = "")
fasta_out_folder <- paste(WordMatchDir, "/FastaOut/Phy_query", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractorGOOD.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             fasta_out_folder,
             sep = " ")
system(cmd)


##############
#            #
# PHY-nsp    #
#            #
##############

#Define your 6 arguments to run the perl script
query_in <- paste(WordMatchDir, "genusOm.txt", sep = "")
otu_in <- paste(pathDataAn, "Phy.nsp.table.tsv", sep = "")
meta_in <- paste(pathDataAn, "Phy.nsp.meta.tsv", sep = "")
fasta_in <- paste(pathDataAn, "new_refseqs_Phynsp.fna", sep = "")
table_taxon_out <- paste(WordMatchDir, "Phynsp_query_by_taxon2.tsv", sep = "")
table_sample_out <- paste(WordMatchDir, "Phynsp_query_by_sample2.tsv", sep = "")
fasta_out_folder <- paste(WordMatchDir, "/FastaOut/Phy_nsp_query", sep = "")

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractorGOOD.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             fasta_out_folder,
             sep = " ")
system(cmd)



```
#cluster data from fasta info from word Search
```{r}


#########################
#                       #
#    ITS by genuses     #
#                       #
#########################

#Define your 2 arguments to run the perl script
args = c("/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/WordMatch/FastaOut/ITS_genus_query", "/home/CFIA-ACIA/tremblaye/fasta_target/fasta_sequences")
analysisFolder <- paste(WordMatchDir, "FastaOut/ITS_genus_query", sep = "")
blastDBfastaPath <- "/home/CFIA-ACIA/tremblaye/fasta_target/fasta_sequences"


cmd <- ("/home/CFIA-ACIA/tremblaye/scripts/ClusterDataFromTaxon3.sh")

system2("/home/CFIA-ACIA/tremblaye/scripts/ClusterDataFromTaxon3.sh", args = args)

cmd <- paste("perl", paste(ScriptsPath, "metaResultExtractorGOOD.pl", sep = ""),
             query_in,
             otu_in,
             meta_in,
             fasta_in,
             table_taxon_out,
             table_sample_out,
             fasta_out_folder,
             sep = " ")
system(cmd)


```
# subset and create abundance tables

```{r}


dir.create(paste(dataAnPath, "abundTables", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

abundTablesPath <- paste(dataAnPath, "abundTables/", sep = "")


##############
#            #
#    ITS     #
#            #
############## 


#Subset OTU tables and meta tables

##############
#            #
#  insects   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

insectITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Insect"), and = FALSE)

write.table(insectITSsubset$data, file=paste(abundTablesPath, "ITS.insectAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(insectITSsubset$meta, file=paste(abundTablesPath, "ITS.insectAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.insectAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.insectAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "p", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

InsectITSmean <- colMeans(test5)
InsectITSmean

##############
#            #
#  Rotorod   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

RotorodITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Rotorod"), and = FALSE)

write.table(RotorodITSsubset$data, file=paste(abundTablesPath, "ITS.RotorodAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(RotorodITSsubset$meta, file=paste(abundTablesPath, "ITS.RotorodAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.RotorodAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.RotorodAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "p", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

RotorodITSmean <- colMeans(test5)
RotorodITSmean

##############
#            #
#    Spore   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"), row.names =1)

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, row.names = 1)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

SporeITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Spore", "Spore_mini"), and = FALSE)

write.table(SporeITSsubset$data, file=paste(abundTablesPath, "ITS.SporeAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = TRUE,
                               quote=FALSE)
  
write.table(SporeITSsubset$meta, file=paste(abundTablesPath, "ITS.SporeAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.SporeAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.SporeAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "p", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

SporeITSmean <- colMeans(test5)
SporeITSmean

##############
#            #
#    Soil   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

SoilITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Soil"), and = FALSE)

write.table(SoilITSsubset$data, file=paste(abundTablesPath, "ITS.SoilAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(SoilITSsubset$meta, file=paste(abundTablesPath, "ITS.SoilAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.SoilAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.SoilAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "p", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

SoilITSmean <- colMeans(test5)
SoilITSmean


#Subset OTU tables and meta tables

##############
#            #
#  Lures     #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)

"alpha_pinene", "C6C8", "General_Longhorn", "Monochamus", "unknown"

metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

alphaPinITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("alpha_pinene"), and = FALSE)

write.table(alphaPinITSsubset$data, file=paste(abundTablesPath, "ITS.alphaPinAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(alphaPinITSsubset$meta, file=paste(abundTablesPath, "ITS.alphaPinAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.alphaPinAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.alphaPinAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

alphaPinITSmean <- colMeans(test5)
alphaPinITSmean



C6C8ITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("C6C8"), and = FALSE)

write.table(C6C8ITSsubset$data, file=paste(abundTablesPath, "ITS.C6C8Abund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(C6C8ITSsubset$meta, file=paste(abundTablesPath, "ITS.C6C8Abund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.C6C8Abund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.C6C8Abund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

C6C8ITSmean <- colMeans(test5)
C6C8ITSmean



General_LonghornITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("General_Longhorn"), and = FALSE)

write.table(General_LonghornITSsubset$data, file=paste(abundTablesPath, "ITS.General_LonghornAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(General_LonghornITSsubset$meta, file=paste(abundTablesPath, "ITS.General_LonghornAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.General_LonghornAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.General_LonghornAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

General_LonghornITSmean <- colMeans(test5)
General_LonghornITSmean


MonochamusITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("Monochamus"), and = FALSE)

write.table(MonochamusITSsubset$data, file=paste(abundTablesPath, "ITS.MonochamusAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(MonochamusITSsubset$meta, file=paste(abundTablesPath, "ITS.MonochamusAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.MonochamusAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.MonochamusAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

MonochamusITSmean <- colMeans(test5)
MonochamusITSmean



unknownITSsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("unknown"), and = FALSE)

write.table(unknownITSsubset$data, file=paste(abundTablesPath, "ITS.unknownAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(unknownITSsubset$meta, file=paste(abundTablesPath, "ITS.unknownAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.unknownAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "ITS.unknownAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

unknownITSmean <- colMeans(test5)
unknownITSmean

##############
#            #
#   Oom      #
#            #
############## 


#Subset OTU tables and meta tables

##############
#            #
#  insects   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

insectOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Insect"), and = FALSE)

write.table(insectOmsubset$data, file=paste(abundTablesPath, "Om.insectAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(insectOmsubset$meta, file=paste(abundTablesPath, "Om.insectAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.insectAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.insectAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

InsectOmmean <- colMeans(test5)
InsectOmmean

##############
#            #
#  Rotorod   #
#            #
############## 

# tabletemp <-  read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"), row.names = 1)
# 
# tabletemp$row.names <- NULL 
#  
# metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE)
# 
# 
# metatemp$row.names <- NULL 
# rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  
# 
# RotorodOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Rotorod"), and = FALSE)
# 
# write.table(RotorodOmsubset$data, file=paste(abundTablesPath, "Om.RotorodAbund.table.tsv", sep = ""),
#                                append    = FALSE,
#                                sep       = "\t",
#                                row.names = FALSE,
#                                quote=FALSE)
#   
# write.table(RotorodOmsubset$meta, file=paste(abundTablesPath, "Om.RotorodAbund.meta.tsv", sep = ""),
#                                append    = FALSE,
#                                sep       = "\t",
#                                row.names = FALSE,
#                                quote=FALSE)
# 
# #read tables
# tabletempsub <-  read.table(paste(abundTablesPath, "Om.RotorodAbund.table.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
# 
# metatempsub <-  read.table(paste(abundTablesPath, "Om.RotorodAbund.meta.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE)
# 
# 
# #metatemp$row.names <- NULL 
# #rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  
# 
# #Get abundance for subseted tables
# test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = TRUE, top = NULL, count = FALSE, mode = "percent")
# 
# #calculate mean of columns
# 
# RotorodOmmean <- colMeans(test5)
# RotorodOmmean

##############
#            #
#    Spore   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

SporeOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Spore", "Spore_mini"), and = FALSE)

write.table(SporeOmsubset$data, file=paste(abundTablesPath, "Om.SporeAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(SporeOmsubset$meta, file=paste(abundTablesPath, "Om.SporeAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.SporeAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.SporeAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

SporeOmmean <- colMeans(test5)
SporeOmmean

##############
#            #
#    Soil   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

SoilOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Soil"), and = FALSE)

write.table(SoilOmsubset$data, file=paste(abundTablesPath, "Om.SoilAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(SoilOmsubset$meta, file=paste(abundTablesPath, "Om.SoilAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.SoilAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.SoilAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "p", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

SoilOmmean <- colMeans(test5)
SoilOmmean


#########
# LURES #
#########


tabletemp <-  read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Om.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)

#"alpha_pinene", "C6C8", "General_Longhorn", "Monochamus", "unknown"

metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

alphaPinOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("alpha_pinene"), and = FALSE)

write.table(alphaPinOmsubset$data, file=paste(abundTablesPath, "Om.alphaPinAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(alphaPinOmsubset$meta, file=paste(abundTablesPath, "Om.alphaPinAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.alphaPinAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.alphaPinAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

alphaPinOmmean <- colMeans(test5)
alphaPinOmmean


C6C8Omsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("C6C8"), and = FALSE)

write.table(C6C8Omsubset$data, file=paste(abundTablesPath, "Om.C6C8Abund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(C6C8Omsubset$meta, file=paste(abundTablesPath, "Om.C6C8Abund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.C6C8Abund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.C6C8Abund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

C6C8Ommean <- colMeans(test5)
C6C8Ommean

General_LonghornOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("General_Longhorn"), and = FALSE)

write.table(General_LonghornOmsubset$data, file=paste(abundTablesPath, "Om.General_LonghornAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(General_LonghornOmsubset$meta, file=paste(abundTablesPath, "Om.General_LonghornAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.General_LonghornAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.General_LonghornAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

General_LonghornOmmean <- colMeans(test5)
General_LonghornOmmean


MonochamusOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("Monochamus"), and = FALSE)

write.table(MonochamusOmsubset$data, file=paste(abundTablesPath, "Om.MonochamusAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(MonochamusOmsubset$meta, file=paste(abundTablesPath, "Om.MonochamusAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.MonochamusAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.MonochamusAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

MonochamusOmmean <- colMeans(test5)
MonochamusOmmean


unknownOmsubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("Lure"), values = c("unknown"), and = FALSE)

write.table(unknownOmsubset$data, file=paste(abundTablesPath, "Om.unknownAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(unknownOmsubset$meta, file=paste(abundTablesPath, "Om.unknownAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Om.unknownAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Om.unknownAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = 10, count = FALSE, mode = "number")

#calculate mean of columns

unknownOmmean <- colMeans(test5)
unknownOmmean


##############
#            #
#   Phy      #
#            #
############## 


#Subset OTU tables and meta tables

##############
#            #
#  insects   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

insectPhysubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Insect"), and = FALSE)

write.table(insectPhysubset$data, file=paste(abundTablesPath, "Phy.insectAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(insectPhysubset$meta, file=paste(abundTablesPath, "Phy.insectAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Phy.insectAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Phy.insectAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = FALSE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

InsectPhymean <- colMeans(test5)
InsectPhymean

##############
#            #
#  Rotorod   #
#            #
############## 
# 
# tabletemp <-  read.table(paste(dataAnPath, "Phy.table.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
# 
# tabletemp$row.names <- NULL 
#  
# metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE)
# 
# 
# metatemp$row.names <- NULL 
# rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  
# 
# RotorodPhysubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Rotorod"), and = FALSE)
# 
# write.table(RotorodPhysubset$data, file=paste(abundTablesPath, "Phy.RotorodAbund.table.tsv", sep = ""),
#                                append    = FALSE,
#                                sep       = "\t",
#                                row.names = FALSE,
#                                quote=FALSE)
#   
# write.table(RotorodPhysubset$meta, file=paste(abundTablesPath, "Phy.RotorodAbund.meta.tsv", sep = ""),
#                                append    = FALSE,
#                                sep       = "\t",
#                                row.names = FALSE,
#                                quote=FALSE)
# 
# #read tables
# tabletempsub <-  read.table(paste(abundTablesPath, "Phy.RotorodAbund.table.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))
# 
# metatempsub <-  read.table(paste(abundTablesPath, "Phy.RotorodAbund.meta.tsv", sep = ""),
#                       sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
#                       as.is = TRUE, check.names = FALSE)
# 
# 
# #metatemp$row.names <- NULL 
# #rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  
# 
# #Get abundance for subseted tables
# test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = TRUE, top = NULL, count = FALSE, mode = "percent")
# 
# #calculate mean of columns
# 
# RotorodPhymean <- colMeans(test5)
# RotorodPhymean

##############
#            #
#    Spore   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

SporePhysubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Spore", "Spore_mini"), and = FALSE)

write.table(SporePhysubset$data, file=paste(abundTablesPath, "Phy.SporeAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(SporePhysubset$meta, file=paste(abundTablesPath, "Phy.SporeAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Phy.SporeAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Phy.SporeAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = TRUE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

SporePhymean <- colMeans(test5)
SporePhymean

##############
#            #
#    Soil   #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "Phy.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

SoilPhysubset <- data.subset(list(data=tabletemp), meta= metatemp, factors = c("TrapType"), values = c("Soil"), and = FALSE)

write.table(SoilPhysubset$data, file=paste(abundTablesPath, "Phy.SoilAbund.table.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
  
write.table(SoilPhysubset$meta, file=paste(abundTablesPath, "Phy.SoilAbund.meta.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "Phy.SoilAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

metatempsub <-  read.table(paste(abundTablesPath, "Phy.SoilAbund.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


#metatemp$row.names <- NULL 
#rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

#Get abundance for subseted tables
test5 <- tax.abund(tabletempsub, otu2=NULL, rank = "g", drop.unclassified = TRUE, top = NULL, count = FALSE, mode = "percent")

#calculate mean of columns

SoilPhymean <- colMeans(test5)
SoilPhymean

```

#percent classified

```{r}

dir.create(paste(dataAnPath, "percentClass", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

PercentClassPath <- paste(dataAnPath, "percentClass/", sep = "")


##############
#            #
#    ITS     #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 

ITS.percent.classified <- percent.classified(data = list(data = tabletemp), ranks = ranks)

write(unlist(ITS.percent.classified), file=paste(PercentClassPath, "ITS.percent.class.ranks.txt", sep = ""))


 
##############
#            #
#    Oom     #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Om.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 

Om.percent.classified <- percent.classified(data = list(data = tabletemp), ranks = ranks)

write(unlist(Om.percent.classified), file=paste(PercentClassPath, "Om.percent.class.ranks.txt", sep = ""))     
      
 
##############
#            #
#    Phy     #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "Phy.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 

Phy.percent.classified <- percent.classified(data = list(data = tabletemp), ranks = ranks)

write(unlist(Phy.percent.classified), file=paste(PercentClassPath, "Phy.percent.class.ranks.txt", sep = ""))     
      

```
#top.group.plot

```{r}
dir.create(paste(dataAnPath, "TopGroupPlot", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

TopGroupPath <- paste(dataAnPath, "TopGroupPlot/", sep = "")

##############
#            #
#    ITS     #
#            #
############## 

#INSECTS

#read tables
tabletempsub <-  read.table(paste(abundTablesPath, "ITS.insectAbund.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))



# Open file-handle to get ready to make the png graph:
  png(filename=paste(TopGroupPath, "ITS", ".top.10.all.ranks.png", sep = ""),
      width = 1500, height = 1000, units = "px")

ITS.top.10 <- group.top.number(list(data=tabletemp), top = 10, ranks = "s", drop.unclassified = FALSE, cex.x = 10, main = "Top 10 at each taxonomic rank in Insect traps")
                 
# Close the png graph file handle:
  dev.off()

```

#plot Venn


```{r}
dir.create(paste(dataAnPath, "VennPlot", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

VennPath <- paste(dataAnPath, "VennPlot/", sep = "")

##############
#            #
#    ITS     #
#            #
############## 

tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

coreTaxaITSTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "TrapType", percent = 0)

taxaInsect <- coreTaxaITSTrapsp$data$Insect$taxa
taxaSoil   <- coreTaxaITSTrapsp$data$Soil$taxa
taxaSpore  <- c(unique(coreTaxaITSTrapsp$data$Spore$taxa, coreTaxaITSTrapsp$data$Spore_mini$taxa))
taxarotorod <- coreTaxaITSTrapsp$data$Rotorod$taxa

taxVector <- list(Insect=taxaInsect, Soil=taxaSoil, Spore=taxaSpore, Rotorod=taxarotorod)


# Open file-handle to get ready to make the png graph:
  png(filename=paste(VennPath, "ITS", ".VennPlot.traps.png", sep = ""),
      width = 1800, height = 1000, units = "px")

group.venn(vectors = taxVector, cat.cex =8, cex = 8, label = FALSE, lab.cex = 8, fill = TRUE)
                 
# Close the png graph file handle:
  dev.off()

  
  
coreTaxaITSLure <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "Lure", percent = 0)

taxaAlphapin <- coreTaxaITSLure$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaITSLure$data$C6C8$taxa
taxaGenLong  <- coreTaxaITSLure$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaITSLure$data$Monochamus$taxa
taxaunknown <- coreTaxaITSLure$data$unknown$taxa
#taxaEthyGlyc <- coreTaxaITSLure$data$Ethylene_glycol$taxa

taxVector <- list(Alpha_pinene=taxaAlphapin, C6C8=taxaC6C8, General_Longhorn=taxaGenLong, Monochamus=taxaMonochamus, Unknown=taxaunknown)


# Open file-handle to get ready to make the png graph:
  png(filename=paste(VennPath, "ITS", ".VennPlot.lures.png", sep = ""),
      width = 5000, height = 4000, units = "px")

group.venn(vectors = taxVector, cat.cex =6, cex = 6, label = FALSE, lab.cex = 6, fill = TRUE)
                 
# Close the png graph file handle:
  dev.off()


  tabletemp <-  read.table(paste(dataAnPath, "ITS.table.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("taxonomy"="character"))

tabletemp$row.names <- NULL 
 
metatemp <-  read.table(paste(pathDataAn, "ITS.meta.tsv", sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)


metatemp$row.names <- NULL 
rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)] #seems to work  

coreTaxaITSTrapsp <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "TrapType", percent = 0)

taxaInsect <- coreTaxaITSTrapsp$data$Insect$taxa
taxaSoil   <- coreTaxaITSTrapsp$data$Soil$taxa
taxaSpore  <- coreTaxaITSTrapsp$data$Spore$taxa
taxarotorod <- coreTaxaITSTrapsp$data$Rotorod$taxa
taxaSporemini <- coreTaxaITSTrapsp$data$Spore_mini$taxa
taxVector <- list(Sporemini=taxaSporemini, Spore=taxaSpore,Insect=taxaInsect, Rotorod=taxarotorod, Soil=taxaSoil)


# Open file-handle to get ready to make the png graph:
  png(filename=paste(VennPath, "ITS", ".VennPlot.traps.with.sporeMini.png", sep = ""),
      width = 2500, height = 2000, units = "px")

group.venn(vectors = taxVector, cat.cex =2.25, cex = 3, label = FALSE, lab.cex = 3, fill = TRUE, lab.col = "pink")
                 
# Close the png graph file handle:
  dev.off()
  
  
  ###
  
  coreTaxaITSLure <- core.Taxa(data = list(data = tabletemp), is.OTU = TRUE, meta = metatemp, rank = "s", drop.unclassified = FALSE, meta.factor = "Lure", percent = 0)

taxaAlphapin <- coreTaxaITSLure$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaITSLure$data$C6C8$taxa
#taxaSGenLong  <- coreTaxaITSLure$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaITSLure$data$Monochamus$taxa
taxaunknown <- coreTaxaITSLure$data$unknown$taxa
#taxaEthyGlyc <- coreTaxaITSLure$data$Ethylene_glycol$taxa

taxVector <- list(Alpha_pinene=taxaAlphapin, C6C8=taxaC6C8, Monochamus=taxaMonochamus, Unknown=taxaunknown)


# Open file-handle to get ready to make the png graph:
  png(filename=paste(VennPath, "ITS", ".VennPlot.lures.no.Genlong.png", sep = ""),
      width = 5000, height = 4000, units = "px")

group.venn(vectors = taxVector, cat.cex =6, cex = 6, label = FALSE, lab.cex = 6, fill = TRUE)
                 
# Close the png graph file handle:
  dev.off()

###
  
#no unknown
 
taxaAlphapin <- coreTaxaITSLure$data$alpha_pinene$taxa
taxaC6C8   <- coreTaxaITSLure$data$C6C8$taxa
taxaGenLong  <- coreTaxaITSLure$data$General_Longhorn$taxa
taxaMonochamus <- coreTaxaITSLure$data$Monochamus$taxa
#taxaunknown <- coreTaxaITSLure$data$unknown$taxa
#taxaEthyGlyc <- coreTaxaITSLure$data$Ethylene_glycol$taxa

taxVector <- list(Alpha_pinene=taxaAlphapin, C6C8=taxaC6C8, General_Longhorn=taxaGenLong, Monochamus=taxaMonochamus)


# Open file-handle to get ready to make the png graph:
  png(filename=paste(VennPath, "ITS", ".VennPlot.lures.no.unknown.png", sep = ""),
      width = 5000, height = 4000, units = "px")

group.venn(vectors = taxVector, cat.cex =6, cex = 6, label = FALSE, lab.cex = 6, fill = TRUE)
                 
# Close the png graph file handle:
  dev.off()
 
  
```