Emilie's script for metagenomics _mod_EMT
========================================================
#Bilodeau mod
```{r}
install.packages("openxlsx")
library(data.table)

# Note, even after Iyad added the dependencies, while it worked in my CFIA-ACIA seesion, 
# coming back to AAFC it required me to get the following packages, so the dependencies 
# installed cluster-wide in the last RedMine were not sufficient, add the following too.
# Should see why there is such issues happening now as well.
install.packages("RAM")
install.packages("caTools")
install.packages("RJSONIO")
install.packages("proto")
install.packages("nnls")
install.packages("quadprog")

```

```{r}
oldPath <- "/home/CFIA-ACIA/tremblaye/2016-07-04_Rstudio_data/"
projectPath <- "/home/CFIA-ACIA/tremblaye/"

# projectName <- "2016-07-04_Rstudio_data/"
projectName <- "NGS_for_Fungi_Detection/"
sharedPath  <- paste(projectPath, projectName, sep = "")
dir.create(paste(projectPath, projectName, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

dataNGS <- "Raw_tables/"
sharedPathData <- paste(sharedPath, dataNGS, sep = "")
dir.create(paste(sharedPath, dataNGS, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

projectPathEm <- "/home/AAFC-AAC/girouxem/"
sharedPathEm  <- paste(projectPathEm, projectName, sep = "")
sharedPathDataEm <- paste(sharedPathEm, dataNGS, sep = "")
pathDataAnEm <- paste(sharedPathEm, "dataAn/", sep = "")
```

*
Setting up the metadata tables - this is to be customized for your data, don't re-run blindly.
```{r}
library("reshape2")
blist1 <- list.files(path = sharedPathDataEm, pattern = glob2rx("*meta*"), full.names=TRUE)
blist2 <- list.files(path = sharedPathDataEm, pattern = glob2rx("*table*"), full.names=TRUE)

metadata1 <- data.frame(blist1)
metadata2 <- data.frame(blist2)

colnames(metadata1)[1] <- "FilePath"
colnames(metadata2)[1] <- "FilePath"

#FileName fixing for metadata files:
metadata1$FileName     <- basename(blist1)
metadata1$ShortName    <- sub("metadata-", "meta_", metadata1$FileName, ignore.case = FALSE)
metadata1$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata1$ShortName, ignore.case = FALSE)
metadata1$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata1$ShortName, ignore.case = FALSE)
metadata1$ShortName    <- sub("-", "_", metadata1$ShortName, ignore.case = FALSE)
parsedCol1             <- data.frame(matrix(unlist(strsplit(as.character(metadata1$ShortName), "_")),
                                            nrow  = length(metadata1$ShortName), 
                                            byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol1)   <- c("DataType", "RunNum", "Primer")
metadata1$ShortName    <- paste(parsedCol1$Primer, parsedCol1$RunNum, parsedCol1$DataType, sep = ".")
metadata1$ShortName    <- tolower(metadata1$ShortName) #put everything in lower cases

#FileName fixing for table files:
metadata2$FileName     <- basename(blist2)
metadata2$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata2$FileName, ignore.case = FALSE)
metadata2$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata2$ShortName, ignore.case = FALSE)
metadata2$ShortName    <- sub("*.otu_table_taxo$|*.otu_table_taxoV2$", "", metadata2$ShortName, ignore.case = FALSE)
metadata2$ShortName    <- sub("^", "table_", metadata2$ShortName, ignore.case = FALSE)
parsedCol2             <- data.frame(matrix(unlist(strsplit(as.character(metadata2$ShortName), "_")),
                                            nrow  = length(metadata2$ShortName), 
                                            byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol2)   <- c("DataType", "Primer", "RunNum")
metadata2$ShortName    <- paste(parsedCol2$Primer, parsedCol2$RunNum, parsedCol2$DataType, sep = ".")
metadata2$ShortName    <- tolower(metadata2$ShortName)

# Combine metadata1 and metadata2:
metadataRaw <- rbind(metadata1, metadata2)
parsedCol   <- data.frame(matrix(unlist(strsplit(as.character(metadataRaw$ShortName), "\\.")), 
                                 nrow  = length(metadataRaw$ShortName), 
                                 byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol) <- c("Primer", "RunNum", "DataType")
metadata            <- cbind(parsedCol[,c(1:3)], metadataRaw)
metadata$ShortName  <- paste(metadata$Primer, metadata$RunNum, sep = ".")
metadata$DirPath    <- paste(sharedPathData)


# Copy the raw data files to an analysis directory, rename by shortname
dir.create(paste(sharedPath, "dataAn/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
pathDataAn <- paste(sharedPath, "dataAn/", sep = "")


for(k in 1:nrow(metadata)){
cmd <- paste("cp ",
             metadata$FilePath[k],
             " ",
             pathDataAn, 
             metadata$ShortName[k], ".", metadata$DataType[k], ".tsv",
             sep = "")
system(cmd)
}

metadata$DataAnFilePath <- pathDataAn
metadata$DataAnFileName <- paste(metadata$ShortName, ".", metadata$DataType, ".tsv", sep = "")

metadataMaster <- dcast(data = metadata,
                        Primer + RunNum + DataAnFilePath + ShortName
                        ~ DataType,
                        value.var = "DataAnFileName",
                        FUN       = c)

write.table(metadataMaster, 
            file      = paste(sharedPath, "My_MetadataTable2.tsv", sep = ""), 
            append    = FALSE,
            quote     = FALSE,
            sep       = "\t", 
            row.names = FALSE)

```
tax.fill to remove uncertae sedis, unclassified...

```{r}
library("RAM")


taxFillDir <- "taxFillTables/"
dir.create(paste(pathDataAn, taxFillDir, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
taxFillPath <- paste(pathDataAn, taxFillDir, sep = "")

i <- 1
#for(i in 1:nrow(metadataMaster)){
for(i in 1:nrow(metadataMaster)){
  temp <-  read.table(paste(metadataMaster$DataAnFilePath[i], metadataMaster$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))
  temp2 <- temp
  temp2$row.names <- NULL #to unactivate numbers generated in the margin
  temp2 <- tax.fill(temp2, downstream=TRUE)
  temp$taxonomy <- temp2$taxonomy
  
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", metadataMaster$ShortName[i], ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE)
}  
 
```
add a column to the metadata file with the tax.fill tables

```{r}
library("reshape2")

taxList <- list.files(path = taxFillPath, pattern = glob2rx("*taxFill.tsv*"), full.names=TRUE)
metaTax <- data.frame(taxList)
colnames(metaTax)[1] <- "taxPath"
metaTax$taxName     <- basename(taxList)


#metaTax$taxPath    <- sub("\\taxFillTables/.*","", metaTax$taxPath, ignore.case = FALSE)
                      sub("\\taxFillTables/.*","\\taxFillTables/",my.data$LANDING)            


#parsedColtax   <- data.frame(matrix(unlist(strsplit(as.character(metadataRaw2$taxPath), sub("taxFillTables/\\S*", "", string))), 
                                 nrow  = length(metadataRaw2$taxPath), 
                                 byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol) <- c("Primer", "RunNum", "DataType")
metadata            <- cbind(parsedCol[,c(1:3)], metadataRaw)
metadata$ShortName  <- paste(metadata$Primer, metadata$RunNum, sep = ".")
metadata$DirPath    <- paste(sharedPathData)


# Copy the raw data files to an analysis directory, rename by shortname

# Combine meta_change and metadataRaw:

meta_change         <- cbind(parsedCol[,c(1)], metadataRaw2)
metadata$ShortName  <- paste(metadata$Primer, metadata$RunNum, sep = ".")
metadata$DirPath    <- paste(sharedPathData)


parsedColtax   <- data.frame(matrix(unlist(strsplit(as.character(meta_change$taxFill), "\\//")), 
                                 nrow  = length(meta_change$TaxFill), 
                                 byrow = TRUE), stringsAsFactors = FALSE)

# Copy the raw data files to an analysis directory, rename by shortname
dir.create(paste(sharedPath, "dataAn/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
pathDataAn <- paste(sharedPath, "dataAn/", sep = "")


for(k in 1:nrow(metadata)){
cmd <- paste("cp ",
             metadata$FilePath[k],
             " ",
             pathDataAn, 
             metadata$ShortName[k], ".", metadata$DataType[k], ".tsv",
             sep = "")
system(cmd)
}

metadata$DataAnFilePath <- pathDataAn
metadata$DataAnFileName <- paste(metadata$ShortName, ".", metadata$DataType, ".tsv", sep = "")

metadataMaster <- dcast(data = metadata,
                        Primer + RunNum + DataAnFilePath + ShortName
                        ~ DataType,
                        value.var = "DataAnFileName",
                        FUN       = c)

write.table(metadataMaster, 
            file      = paste(sharedPath, "My_MetadataTable2.tsv", sep = ""), 
            append    = FALSE,
            quote     = FALSE,
            sep       = "\t", 
            row.names = FALSE)

```

Ecrire ta table de meta (tracabilitÃ© man!)
```{r}
i = 1
for(i in 1:length(meta)){
  df<- read.meta(file=(blist2[i]), sep="\t")  
  write.table(df, file = paste(meta[i], "_b", sep = ""), sep="\t")
}

blist2 <- basename(list.files(pattern="_meta_b", full.names=TRUE))

i = 1
for(i in 1:length(meta)){
  #df<- tax.fill(data=get(table[i]), downstream=TRUE)   
  write.table(get(meta[i]), file = paste(meta[i], "_b", sep = ""), sep="\t")
  #return(df)
}
```


Next:
```{r}
i = 1
for(i in 1:length(clist)){
  temp <- list(clist[i])  
  write.table(temp, file = file.path(paste(workingPath, clist[i], "_df", sep = "")))
}

```



```{r}
dfList <- lapply(clist, function(i) {
  df <- list(data=clist)

})

df <- ldply(clist, data.frame)

dfList <- lapply(clist, function(i) {
  df <- list(i) 
  })

clist <- (list.files(path=workingPath, pattern="_table_b_c", full.names=TRUE))
```

```{r}
data1f=list(data=ITS1F_run1_table)
meta.diversity.run1.its1f<-OTU.diversity(data=data1f, meta=ITS1F_run1_meta)

for (i in 1:length(clist)) assign(clist[i], read.table(clist[i], sep="\t"))

i = 1
for(i in 1:length(blist)){
  temp <- list(data=get(blist[i]))   
 #temp <- setNames(temp, clist)#file = file.path(paste(workingPath, table2[i], ".tsv", sep = "")),
              #append = FALSE, sep = "\t", col.names = NA, quote = FALSE)
  #print(cmd)
}

temp <- setNames(temp, clist)

clist = basename(grep("_table_b_c", grep("run", value=TRUE, ls(all.names=TRUE)), value=TRUE))

clist <- list.files(path = workingPath, pattern = glob2rx("_"), full.names=TRUE)

files <- list.files(path = workingPath, pattern = glob2rx("_table_b"), full.names=TRUE)
```


```{r}
tblecList <- list.files(pattern="*table_b_c")

i = 1
for(i in 1:length(tblecList)){
  df2<- read.OTU(tblecList[i], sep="\t")   
  #dfList <- list(file = paste(tblecList[i], "_df", sep = ""))
  #return(df)
}

salut <- list(salut='ITS1F_run12_table_b_c')

temp = list.files(pattern="*_table.tsv")
for (i in 1:length(list)) assign(list[i], read.table(list[i], sep="\t", quote="", header=TRUE, as.is=FALSE, row.names=NULL))

```
#List your datasets for diversity index. This adds a column in a new diversity metadata file
```{r}



i = 1
for(i in 1:length(blist)){
  df<- read.OTU(file=get(blist[i]), sep="\t")   
  write.table(df, file = paste(blist[i], "_c", sep = ""), sep="\t")
  return(df)
}

data1f=list(data=ITS1F_run1_table)
meta.diversity.run1.its1f<-OTU.diversity(data=data1f, meta=ITS1F_run1_meta)

```
