Emilie's script for metagenomics 
========================================================
```{r}
install.packages("openxlsx")
library(data.table)

# Note, even after Iyad added the dependencies, while it worked in my CFIA-ACIA seesion, 
# coming back to AAFC it required me to get the following packages, so the dependencies 
# installed cluster-wide in the last RedMine were not sufficient, add the following too.
# Should see why there is such issues happening now as well.
install.packages("RAM")
install.packages("caTools")
install.packages("RJSONIO")
install.packages("proto")
install.packages("nnls")
install.packages("quadprog")

```

```{r}
oldPath <- "/home/CFIA-ACIA/tremblaye/2016-07-04_Rstudio_data/"
projectPath <- "/home/CFIA-ACIA/tremblaye/"

# projectName <- "2016-07-04_Rstudio_data/"
projectName <- "NGS_for_Fungi_Detection/"
sharedPath  <- paste(projectPath, projectName, sep = "")
dir.create(paste(projectPath, projectName, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

dataNGS <- "Raw_tables/"
sharedPathData <- paste(sharedPath, dataNGS, sep = "")
dir.create(paste(sharedPath, dataNGS, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

```

*
Setting up the metadata tables - this is to be customized for your data, don't re-run blindly.
```{r}
library("reshape2")

2016-10-28
library("reshape2")
blist1 <- list.files(path = pathDataAn, pattern = glob2rx("*meta*"), full.names=TRUE)
blist2 <- list.files(path = pathDataAn, pattern = glob2rx("*table*"), full.names=TRUE)

metadata1 <- data.frame(blist1)
metadata2 <- data.frame(blist2)
metadata3 <- data.frame(dirname(blist1))

colnames(metadata1)[1] <- "FilePath"
colnames(metadata2)[1] <- "FilePath"
colnames(metadata3)[1] <- "FilePath"

View(metadata1)
View(metadata2)
View(metadata3)

metadata3$meta     <- basename(blist1)
metadata3$table     <- basename(blist2)
metadata3$ShortName    <- sub("*.meta.tsv$ | *.meta.tsv$", "", metadata3$meta, ignore.case = FALSE) #ran later with new seq runs (13-15)


metadata3$Primer   <- gsub("\\..*","", metadata3$ShortName)
View(metadata3)
metadata3$RunNum   <- gsub("^.*\\.","", metadata3$ShortName)
#add a / to your FilePath

metadata3$FilePath <- sub("$", "/", metadata3$FilePath )


older by EmGi



#FileName fixing for metadata files:
metadata1$FileName     <- basename(blist1)
metadata1$ShortName    <- sub("metadata-", "meta_", metadata1$FileName, ignore.case = FALSE)
metadata1$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata1$ShortName, ignore.case = FALSE)
metadata1$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata1$ShortName, ignore.case = FALSE)
metadata1$ShortName    <- sub("-", "_", metadata1$ShortName, ignore.case = FALSE)


metadata1$FileName     <- basename(blist1)
metadata1$ShortName    <- sub("*.tsv$", "", metadata1$FileName, ignore.case = FALSE) #ran later with new seq runs (13-15)

parsedCol1             <- data.frame(matrix(unlist(strsplit(as.character(metadata1$ShortName), "_")),
                                            nrow  = length(metadata1$ShortName), 
                                            byrow = TRUE), stringsAsFactors = FALSE)
colnames(metadata1)   <- c("DataType", "RunNum", "Primer")
metadata1$ShortName    <- paste(parsedCol1$Primer, parsedCol1$RunNum, parsedCol1$DataType, sep = ".")
metadata1$ShortName    <- tolower(metadata1$ShortName) #put everything in lower cases

#FileName fixing for table files:
metadata2$FileName     <- basename(blist2)
metadata2$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata2$FileName, ignore.case = FALSE)
metadata2$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata2$ShortName, ignore.case = FALSE)
metadata2$ShortName    <- sub("*.otu_table_taxo$|*.otu_table_taxoV2$", "", metadata2$ShortName, ignore.case = FALSE)
metadata2$ShortName    <- sub("^", "table_", metadata2$ShortName, ignore.case = FALSE)

metadata2$FileName     <- basename(blist2)
metadata2$ShortName    <- sub("*.tsv$", "", metadata2$FileName, ignore.case = FALSE) #ran later with new seq runs (13-15)

parsedCol2             <- data.frame(matrix(unlist(strsplit(as.character(metadata2$ShortName), "_")),
                                            nrow  = length(metadata2$ShortName), 
                                            byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol2)   <- c("DataType", "Primer", "RunNum")
metadata2$ShortName    <- paste(parsedCol2$Primer, parsedCol2$RunNum, parsedCol2$DataType, sep = ".")
metadata2$ShortName    <- tolower(metadata2$ShortName)

# Combine metadata1 and metadata2:
metadataRaw <- rbind(metadata1, metadata2)
parsedCol   <- data.frame(matrix(unlist(strsplit(as.character(metadataRaw$ShortName), "\\.")), 
                                 nrow  = length(metadataRaw$ShortName), 
                                 byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol) <- c("Primer", "RunNum", "DataType")
metadata            <- cbind(parsedCol[,c(1:3)], metadataRaw)
metadata$ShortName  <- paste(metadata$Primer, metadata$RunNum, sep = ".")
metadata$DirPath    <- paste(sharedPathData)


# Copy the raw data files to an analysis directory, rename by shortname
dir.create(paste(sharedPath, "dataAn/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# capture path to new directory:
dataAnPath  <- paste(sharedPath, "dataAn/", sep = "")



for(k in 1:nrow(metadata)){
cmd <- paste("cp ",
             metadata$FilePath[k],
             " ",
             dataAnPath, 
             metadata$ShortName[k], ".", metadata$DataType[k], ".tsv",
             sep = "")
system(cmd)
}

metadata$DataAnFilePath <- dataAnPath
metadata$DataAnFileName <- paste(metadata$ShortName, ".", metadata$DataType, ".tsv", sep = "")

metadataMaster <- dcast(data = metadata,
                        Primer + RunNum + DataAnFilePath + ShortName
                        ~ DataType,
                        value.var = "DataAnFileName",
                        FUN       = c)


fungi_detection <- "/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection//"



write.table(metadata3, 
            file      = paste(fungi_detection, "metadataMaster_allruns.tsv", sep = ""), 
            append    = FALSE,
            quote     = FALSE,
            sep       = "\t", 
            row.names = FALSE)

```
tax.fill to remove uncertae sedis, unclassified...

```{r}
library("RAM")

taxFillDir <- "taxFillTables/"
dir.create(paste(pathDataAn, taxFillDir, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
taxFillPath <- paste(pathDataAn, taxFillDir, sep = "")

i <- 1
#for(i in 1:nrow(metadataMaster)){
for(i in 1:nrow(metadataMaster)){
  temp <-  read.table(paste(metadataMaster$DataAnFilePath[i], metadataMaster$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))
  temp2 <- temp
  temp2$row.names <- NULL #to unactivate numbers generated in the margin
  # or... row.names(combinedTableOm) <- NULL, becuase we checked the above and it's not what we want - it's not reliable, and the numbers in the margins in previous tables were not removed - HOWEVER, I think the numbers in this section are more pronounced and different - so we need to check before changing this line.
  temp2 <- tax.fill(temp2, downstream=TRUE)
  temp$taxonomy <- temp2$taxonomy
  
  write.table(temp, file=paste(dataAnPath, "taxFillTables/", metadataMaster$ShortName[i], ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)
}  
 #this is rerunning the whole runs including 13-15
i <- 1

for(i in 1:nrow(metadata3)){
  temp <-  read.table(paste(metadata3$FilePath[i], metadata3$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))
  temp2 <- temp
  temp2$row.names <- NULL #to unactivate numbers generated in the margin
  temp2 <- tax.fill(temp2, downstream=TRUE)
  temp$taxonomy <- temp2$taxonomy
  
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", metadata3$ShortName[i], ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)
}  
 

```
Add 2 columns to the metadata file with the tax.fill tables names and paths (steps 1-6)
1-make a list of the filenames

```{r}
library("reshape2")
taxList <- list.files(path = taxFillPath, pattern = glob2rx("*taxFill.tsv*"), full.names=TRUE)
```
2-Tell R it is a dataframe

```{r}
metaTax2 <- data.frame(taxList)
```
3-Give header name to the column being built

```{r}
colnames(metaTax2)[1] <- "taxPath"
```
4-Build a 2nd colum with file name only 

```{r}
metaTax2$taxName     <- basename(taxList)
```
5-Cut basename from the path name

```{r}
metaTax2$taxPath <- sub('([^//]+\\//).*', '\\1', metaTax2$taxPath)
#Here, we match all characters that are not ? ([^?]+) followed by ? (\\?) and use parentheses to capture as a group (([^?]+\\?)), and we leave the rest of characters not in the group (./*). (http://stackoverflow.com/questions/10150579/adding-a-column-to-a-data-frame)
```
6-Remove last /
```{r}
metaTax2$taxPath <- substr(metaTax2$taxPath, 1, nchar(metaTax2$taxPath)-1) 
```
Add-up your two new columns (taxPath and taxName) to the end of the metadata file
```{r}
metadata3$taxPath     <- metaTax2$taxPath
metadata3$taxName     <- metaTax2$taxName
```
write the precious table you just made 
```{r}
write.table(metadata3, 
            file      = paste(fungi_detection, "metadataMasterTax2.tsv", sep = ""), 
            append    = FALSE,
            quote     = FALSE,
            sep       = "\t", 
            row.names = FALSE)

metadataMasterNew <- metadata3
```

Create diversity directory and path. List your datasets for diversity indexes. Adds a bunch of columns in a new diversity metadata file.
Indices obtained (in this specific order) are: Spec Number,  Simpson data,	Inv simpson data,	Shannon data,	Simpson eveness,	Shannon eveness,	Simpson true diversity,	shannon true diversity,	chao,	ACE.

```{r}
diversityDir <- paste(pathDataAn, "diversityDir/", sep ="")
dir.create(paste(pathDataAn, diversityDir, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
diversityPath <- paste(pathDataAn, diversityDir, sep = "")


i <- 16
for(i in 1:nrow(metadataMasterNew)) {
  metatemp <- read.table(paste(metadataMasterNew$FilePath[i], metadataMasterNew$meta[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)
 
  metatemp$row.names <- NULL

  tabletemp <-  read.table(paste(metadataMasterNew$FilePath[i], metadataMasterNew$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
  
  tabletemp$row.names <- NULL
  
  rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)]
   
  temp2 <- OTU.diversity(data = list(data = tabletemp), meta = metatemp)
  write.table(temp2, file=paste(dataAnPath, "diversityDir/", metadataMasterNew$ShortName[i], ".meta.div.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)
  }


```
Create summary barplot and stats table about your OTU. Never use the taxFilled tables as it does not work.

```{r}
library("RAM")
# make new directory:
dir.create(paste(dataAnPath, "otuRecap/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture the path of the new directory:
otuDataPath <- paste(sharedPath, dataAnDir, "otuRecap/", sep = "")

# Create a new directory in otuRecap for plots:
dir.create(paste(otuDataPath, "Plots/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

# Capture path to new directory:
plotPath <- paste(otuDataPath, "Plots/", sep = "")

# Create a new directory in otuRecap for tables:
dir.create(paste(otuDataPath, "StatTables/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
statOTURecapPath <- paste(otuDataPath, "StatTables/", sep = "")

i <- 1
for(i in 1:nrow(metadataMasterTax)){
  temp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  temp$row.names <- NULL 

# Open file-handle to get ready to make the png graph:
  png(filename=paste(dataAnPath/otuRecap/Plots, metadataMasterTax$ShortName[i], ".otuRecap.png", sep = ""),
      width = 800, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(data = list(data = temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Set3")

# Close the png graph file handle:
  dev.off()

# Capture the data also in a dataframe:
  df <- OTU.recap(data = list(data = temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Greens")

# Write out the dataframe:
  write.table(df, file=paste(statOTURecapPath, metadataMasterTax$ShortName[i], ".otuRecap.tsv", sep = ""),
                               # "otuRecap/its1f.run7.otuRecap.tsv", sep = ""),
                               # metadataMasterTax$ShortName[9], ".otuRecap.tsv", sep = "",
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
}

```

### group.abund.Taxa
#### top taxa at each rank (all ranks)
  + Needs more than one site in the OTU table to work properly.
  + top taxa at each rank (all ranks). In number or percent. Moustache plot.
  + Needs more than one site in the OTU table to work properly.
  + default top 10%
```{r}
library("RAM")
# make new directory:
dir.create(paste(pathDataAn, "groupTaxaBar/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
groupTaxaPath <- paste(pathDataAn, "groupTaxaBar/", sep = "")

i <- 1
for(i in 1:nrow(metadataMasterTax)){
  tabletemp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  tabletemp$row.names <- NULL 
  
   metatemp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$meta[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
  
#   rownames(temp) <- colnames(trans)[-ncol(trans)]
# In the following, the column names in the metadata table are put in as the row names
# for the taxa table. Recall, in the metadata table, the species infor
  #rownames(tabletemp) <- colnames(metatemp)[-ncol(metatemp)]
     
# Open file-handle to get ready to make the png graph:
  png(filename=paste(groupTaxaPath, metadataMasterTax$ShortName[i], ".groupTaxBar_species.png", sep = ""),
      width = 800, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
#   group.Taxa.bar(data = list(data = trans), is.OTU = TRUE, func = "sum", rank="k", meta = temp, meta.factor = "TrapType", col.pal = "fullcolors")
group.Taxa.bar(data = list(data = tabletemp), is.OTU = TRUE, func = "sum", rank="k", meta = metatemp, meta.factor = "TrapType", col.pal = "Set3")

# Close the png graph file handle:
  dev.off()

}
```

# The good one!
### group.abund.Taxa
#### top taxa at each rank (all ranks)
  + Needs more than one site in the OTU table to work properly.
  + top taxa at each rank (all ranks). In number or percent. Moustache plot.
  + Needs more than one site in the OTU table to work properly.
  + default top 10%

# 1. generate the list of taxa desired:
```{r}
library("RAM")


# make new directory:
dir.create(paste(pathDataAn, "groupTaxaBar/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
# Capture path to new directory:
groupTaxaPath <- paste(pathDataAn, "groupTaxaBar/", sep = "")


i <- 1
for(i in 1:nrow(metadataMasterTax)){
  tabletemp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
  tabletemp$row.names <- NULL 
  
  metatemp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$meta[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
  metatemp$row.names <- NULL 
  
# In the following, the column names for samples in the otu table become the rownames in the metadata table, which
# should already be the case (but then the row names get removed in metatemp$row.names <- NULL )
  rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)]


taxa <- core.Taxa(data = list(data = tabletemp),
                  is.OTU = TRUE,
                  rank = "f",    # Pay attention to the rank you want to work on
                  drop.unclassified = TRUE,
                  meta = metatemp,
                  meta.factor = "TrapType",
                  percent = 0.7) # Pay attention to the percent, you 
                                 # will retrieve those taxa that are present 
                                 # in that fraction at that rank.
taxaInsect <- taxa$data$Insect$taxa
#taxaInsect <- taxa$data$Insect$summary
taxaSoil   <- taxa$data$Soil$taxa
taxaSpore  <- taxa$data$Spore$taxa
taxaAll    <- unique(c(taxaInsect, taxaSoil, taxaSpore))
```

```{r}

i <- 1
for(i in 1:nrow(metadataMasterTax)){
  tabletemp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character", "taxonomy"="character"))
  
  tabletemp$row.names <- NULL 
  
  metatemp <-  read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$meta[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
  metatemp$row.names <- NULL 
  
# In the following, the column names for samples in the otu table become the rownames in the metadata table, which
# should already be the case (but then the row names get removed in metatemp$row.names <- NULL )
  rownames(metatemp) <- colnames(tabletemp)[-ncol(tabletemp)]

# Open file-handle to get ready to make the png graph:
  png(filename=paste(groupTaxaPath, metadataMaster$ShortName[i], ".groupTaxBar_species_Trap.png", sep = ""),
      width = 800, height = 500, units = "px")

# The following is for the function group.taxa.bar. This function doesn't work for us.
# As an alternative to this function, Wen's pdf offers group.abun.Taxa, immediately following
# this code.
# group.Taxa.bar(#data = tabletemp,
#                data = list(data = tabletemp), 
#                is.OTU = TRUE, 
#                func = "sum", 
#                rank="g__", 
#                #taxa = tabletemp,
#                taxa = taxaAll,
#                meta = metatemp, 
#                meta.factor = "TrapType", 
#                col.pal = "Set3")

# The following is an alternative to group.taxa.bar:

group.abund.Taxa(data = list(data = tabletemp), 
                 is.OTU = TRUE,
                 # taxa = tabletemp,
                 taxa = taxaAll,
                 drop.unclassified = TRUE,
                 bar.width = NULL,
                 meta = metatemp,
                 meta.factor = "TrapType",
                 RAM.theme = NULL,
                 col.pal = NULL,
                 main = "",
                 file = NULL,
                 ext = NULL,
                 height = 50,
                 width = 16)

# Close the png graph file handle:
  dev.off()

}
```


# Produce the rarefaction curve for each sample (copied from Wen)

# First transpose the otu table, because the rarecurve take each row as one sample

```{r}
TransposedOTUTables <- "transposed_OTU_tables/"
dataAnPath <- "/home/CFIA-ACIA/tremblaye/NGS_for_Fungi_Detection/dataAn/"
dir.create(paste(dataAnPath, TransposedOTUTables, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
TransposedOTUTables <- paste(dataAnPath, TransposedOTUTables, sep = "")

in linux: cp *.table.tsv transposed_OTU_tables/


i <- 1
for(i in 1:nrow(metadataMasterTax)){
  temp <-  read.table(paste(metadataMaster$DataAnFilePath[i], metadataMasterTax$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE, colClasses=c("row.names"="character"))
  temp$row.names <- NULL 
  temp <- as.data.frame(temp)
  
  # Capture the data also in a dataframe:
  temp <- transpose.OTU(data = list(temp))
  temp <- transpose.OTU(metadataMasterTax$table[i])
  # Write out the dataframe:
  write.table(temp, file=paste(TransposedOTUTables, metadataMasterTax$ShortName[i], ".transposed.tsv", sep = ""),
                               append    = FALSE,
                               sep       = "\t",
                               row.names = FALSE,
                               quote=FALSE)
}


# Open file-handle to get ready to make the raref curve:
  png(filename=paste(dataAnPath/trans, metadataMasterTax$ShortName[i], ".otuRecap.png", sep = ""),
      width = 800, height = 500, units = "px")

# Put this data into the png graph for the open filehandle:
  OTU.recap(data = list(data = temp), ranks=c("k", "p", "c", "o", "f", "g", "s"), brewer.pal="Set3")

# Close the png graph file handle:
  dev.off()




```

```{r}
ITS1.t<-t(ITS1[, -dim(ITS1)[2]])
head(ITS1.t)[,1:5]
ITS2.t<-t(ITS2[, -dim(ITS2)[2]])
head(ITS2.t)[,1:5]

##from forum##
library(reshape2)
dcast.data.table(melt(mydata, id.vars = "col0"), variable ~ col0)
###
ab<-table(unlist(ITS1.t))
ab
x11(width = 12, height = 4)
# do a barplot of frequency of abundance ranks
barplot(log(ab), las=1, xlab = "Abundand in matrix", ylab = "log (frequency)", ylim = c(0, max(log(ab))+2))

# Total count/proportion of absences
sum(ITS1.t == 0)
sum(ITS2.t == 0)
sum(ITS1.t == 0)/(nrow(ITS1.t)*ncol(ITS1.t))
sum(ITS2.t == 0)/(nrow(ITS2.t)*ncol(ITS2.t))

```

# plot rarecurve for each sample

```{r}
library("vegan")
#tiff(compression="lzw", file=paste(basename, sep="", "ITS.rarecurve.tiff"), res=1000, units="in", width=16, height=8)

x11(width=8, height=4)
par(mfrow=c(1,2))
rarecurve(ITS1.t,step=20, sample=min(rowSums(ITS1.t)), col="blue", cex=0.6, xlab="Number of sequences", ylab="Number of OTUs", main="ITS1")
rarecurve(ITS2.t,step=20, sample=min(rowSums(ITS2.t)), col="blue", cex=0.6, xlab="Number of sequences", ylab="Number of OTUs", main="ITS2")
dev.off()

#calculate species accumulation curves for subsets of the community and environmental data sets
library(BiodiversityR)

x11(width=8, height=4)
par(mfrow=c(1,2))
accumcomp(ITS1.t, y=meta, factor='Crop', method="rarefaction",cex=0.5, ylab="OTU number", xlab="Samples") # need to place the lengend by click on the plot before moving to next plot
accumcomp(ITS1.t, y=meta, factor='City', method="rarefaction",cex=0.5, ylab="OTU number", xlab="Samples")
dev.off()

#dev.copy(png, file="ITS1.otu.rarecurve.eachsample.png")

#dev.off()
```

Setting the data up so that we generate metadata tables specific to primer set.
The metadata table, metadataMasterNew, has all the information for the tables and metatables
for all sequencing runs.
1. We will use metadataMasterNew to make subsest tables for each primer set.
2. We will 

Combine metatables by primer set:
```{r}

comColNames <- c("Primer", "ShortName", "FilePath", "meta", "table", "taxPath", "taxName")

combinedTableITS <- subset(metadataMasterNew, 
                           metadataMasterNew$Primer == "its1f" | metadataMasterNew$Primer == "its2", 
                           select = c(comColNames))
row.names(combinedTableITS) <- NULL
combinedTableITS$tablePath  <- paste(combinedTableITS$FilePath, combinedTableITS$table, sep = "")
combinedTableITS$metaPath   <- paste(combinedTableITS$FilePath, combinedTableITS$meta, sep = "")
combinedTableITS$taxFillPath  <- paste(combinedTableITS$taxPath, combinedTableITS$taxName, sep = "")

combinedTableOm <- subset(metadataMasterNew, 
                          metadataMasterNew$Primer == "omlo" | metadataMasterNew$Primer == "omup", 
                          select = c(comColNames))
row.names(combinedTableOm) <- NULL
combinedTableOm$tablePath  <- paste(combinedTableOm$FilePath, combinedTableOm$table, sep = "")
combinedTableOm$metaPath   <- paste(combinedTableOm$FilePath, combinedTableOm$meta, sep = "")
combinedTableOm$taxFillPath  <- paste(combinedTableOm$taxPath, combinedTableOm$taxName, sep = "")


combinedTablePhy <- subset(metadataMasterNew, 
                           metadataMasterNew$Primer == "phy", 
                           select = c(comColNames))
row.names(combinedTablePhy) <- NULL
combinedTablePhy$tablePath  <- paste(combinedTablePhy$FilePath, combinedTablePhy$table, sep = "")
combinedTablePhy$metaPath   <- paste(combinedTablePhy$FilePath, combinedTablePhy$meta, sep = "")
combinedTablePhy$taxFillPath  <- paste(combinedTablePhy$taxPath, combinedTablePhy$taxName, sep = "")
```


Read in each meta table from a list, and rbind the list while calling specific columns by name.
```{r}
library(data.table)

metaToAppend <- combinedTableITS$metaPath
metaColToKeep <- c("row.names", "BarcodeSequence", "LinkerPrimerSequence", 
                   "InputFileName", "TrapType", "Lure",
                   "CollectionDa  te", "Province", "City", "Description" )

metaTemp <- rbindlist(lapply(metaToAppend, fread,
                            header = "auto",
                            select = metaColToKeep))

```

```{r}
library(data.table)

i <- 1

for(i in 1:nrow(combinedTableITS)){
   meta_ITS_merged_temp <- fread(combinedTableITS$metaPath[i],
                      select = c("row.names", "BarcodeSequence", "LinkerPrimerSequence", "InputFileName", "TrapType", "Lure",
                                 "CollectionDa  te", "Province", "City", "Description" ))
   
   write.table(meta_ITS_merged_temp, file=paste(pathDataAn, "meta_ITS_merged", ".tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote = FALSE)
              
    }

meta_ITS_merged <- fread("combinedTableITS", select = c("row.names", "BarcodeSequence", "LinkerPrimerSequence", "InputFileName", "TrapType", "Lure", "CollectionDate", "Province", "City", "Description" ))
 
combine.OTU(data=list(data=combinedTableITS$tablePath), meta=list(data=combinedTableITS$metaPath))




i <- 1

for(i in 1:nrow(combinedTableITS)){
  tabletemp <-  read.table(combinedTableITS$tablePath[i],
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))
  
  metatemp <-  read.table(combinedTableITS$metaPath[i],
                      sep = "\t", header = TRUE, dec = ".", 
                      comment.char = "", 
                      quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)


  tabletemp$row.names <- NULL
  # row.names(combinedTableOm) <- NULL
  
  test<- combine.OTU(data=list(tabletemp), meta=metatemp)
  is.data.frame(test)
  }
  
  #write.table(test, file=paste(dataAnPath, "taxFillTables/", metadataMaster$ShortName[i], ".table.taxFill.tsv", sep = ""),
              #append    = FALSE,
              #sep       = "\t",
#               row.names = FALSE,
#               quote = FALSE)
}  
 

```

```{r}

filenames=list.files(path=myPath, 
                   pattern = "^its1f.*meta.tsv$|^its2.*meta.tsv$",
                   recursive = FALSE,
                   full.names=TRUE)
for (file in filenames){
  # if the merged dataset doesn't exist, create it
  if (!exists("dataset")){
    dataset <- read.table(file, header=TRUE, sep="\t")
  }
  # if the merged dataset does exist, append to it
  if (exists("dataset")){
    temp_dataset <-read.table(file, header=TRUE, sep="\t")
    dataset<-rbind(dataset, temp_dataset)
    rm(temp_dataset)
  }
}

metacolumns <- c("row.names", "BarcodeSequence", "LinkerPrimerSequence", "InputFileName", "TrapType", "Lure", 
  "CollectionDate", "Province", "City", "Description" )

k <- 1
for(k in names(filenames)){
  if (!exists("tempMeta")){
  tempMeta <- read.table(filenames[k], 
                         header = TRUE, 
                         sep = "\t",
                         dec = ".", 
                         comment.char = "", 
                         quote = "", 
                         stringsAsFactors = TRUE,
                         as.is = TRUE, 
                         check.names = FALSE)
  tempMeta <- tempMeta[,c(metacolumns)]
  }
  # if the merged dataset does exist, append to it
  if (exists("tempMeta")){
  temp_tempMeta <-read.table(filenames[k], 
                        header = TRUE, 
                        sep = "\t",
                        dec = ".", 
                        comment.char = "", 
                        quote = "", 
                        stringsAsFactors = TRUE,
                        as.is = TRUE, 
                        check.names = FALSE)
  temp_tempMeta <- temp_tempMeta[,c(metacolumns)]
  
  tempMeta <-rbind(tempMeta, temp_tempMeta)
  rm(temp_tempMeta)
} 
}


```
