Emilie's script for metagenomics _mod_EMT
========================================================
#Bilodeau mod
```{r}
install.packages("openxlsx")
library(data.table)

# Note, even after Iyad added the dependencies, while it worked in my CFIA-ACIA seesion, 
# coming back to AAFC it required me to get the following packages, so the dependencies 
# installed cluster-wide in the last RedMine were not sufficient, add the following too.
# Should see why there is such issues happening now as well.
install.packages("RAM")
install.packages("caTools")
install.packages("RJSONIO")
install.packages("proto")
install.packages("nnls")
install.packages("quadprog")

```

```{r}
oldPath <- "/home/CFIA-ACIA/tremblaye/2016-07-04_Rstudio_data/"
projectPath <- "/home/CFIA-ACIA/tremblaye/"

# projectName <- "2016-07-04_Rstudio_data/"
projectName <- "NGS_for_Fungi_Detection/"
sharedPath  <- paste(projectPath, projectName, sep = "")
dir.create(paste(projectPath, projectName, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

dataNGS <- "Raw_tables/"
sharedPathData <- paste(sharedPath, dataNGS, sep = "")
dir.create(paste(sharedPath, dataNGS, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)

projectPathEm <- "/home/AAFC-AAC/girouxem/"
sharedPathEm  <- paste(projectPathEm, projectName, sep = "")
sharedPathDataEm <- paste(sharedPathEm, dataNGS, sep = "")
pathDataAnEm <- paste(sharedPathEm, "dataAn/", sep = "")
```

*
Setting up the metadata tables - this is to be customized for your data, don't re-run blindly.
```{r}
library("reshape2")
blist1 <- list.files(path = sharedPathDataEm, pattern = glob2rx("*meta*"), full.names=TRUE)
blist2 <- list.files(path = sharedPathDataEm, pattern = glob2rx("*table*"), full.names=TRUE)

metadata1 <- data.frame(blist1)
metadata2 <- data.frame(blist2)

colnames(metadata1)[1] <- "FilePath"
colnames(metadata2)[1] <- "FilePath"

#FileName fixing for metadata files:
metadata1$FileName     <- basename(blist1)
metadata1$ShortName    <- sub("metadata-", "meta_", metadata1$FileName, ignore.case = FALSE)
metadata1$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata1$ShortName, ignore.case = FALSE)
metadata1$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata1$ShortName, ignore.case = FALSE)
metadata1$ShortName    <- sub("-", "_", metadata1$ShortName, ignore.case = FALSE)
parsedCol1             <- data.frame(matrix(unlist(strsplit(as.character(metadata1$ShortName), "_")),
                                            nrow  = length(metadata1$ShortName), 
                                            byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol1)   <- c("DataType", "RunNum", "Primer")
metadata1$ShortName    <- paste(parsedCol1$Primer, parsedCol1$RunNum, parsedCol1$DataType, sep = ".")
metadata1$ShortName    <- tolower(metadata1$ShortName) #put everything in lower cases

#FileName fixing for table files:
metadata2$FileName     <- basename(blist2)
metadata2$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata2$FileName, ignore.case = FALSE)
metadata2$ShortName    <- sub("*.tsv$|*.tab.txt$|*.tsv.txt$|*.tab$|*.tabv2.txt$", "", metadata2$ShortName, ignore.case = FALSE)
metadata2$ShortName    <- sub("*.otu_table_taxo$|*.otu_table_taxoV2$", "", metadata2$ShortName, ignore.case = FALSE)
metadata2$ShortName    <- sub("^", "table_", metadata2$ShortName, ignore.case = FALSE)
parsedCol2             <- data.frame(matrix(unlist(strsplit(as.character(metadata2$ShortName), "_")),
                                            nrow  = length(metadata2$ShortName), 
                                            byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol2)   <- c("DataType", "Primer", "RunNum")
metadata2$ShortName    <- paste(parsedCol2$Primer, parsedCol2$RunNum, parsedCol2$DataType, sep = ".")
metadata2$ShortName    <- tolower(metadata2$ShortName)

# Combine metadata1 and metadata2:
metadataRaw <- rbind(metadata1, metadata2)
parsedCol   <- data.frame(matrix(unlist(strsplit(as.character(metadataRaw$ShortName), "\\.")), 
                                 nrow  = length(metadataRaw$ShortName), 
                                 byrow = TRUE), stringsAsFactors = FALSE)
colnames(parsedCol) <- c("Primer", "RunNum", "DataType")
metadata            <- cbind(parsedCol[,c(1:3)], metadataRaw)
metadata$ShortName  <- paste(metadata$Primer, metadata$RunNum, sep = ".")
metadata$DirPath    <- paste(sharedPathData)


# Copy the raw data files to an analysis directory, rename by shortname
dir.create(paste(sharedPath, "dataAn/", sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
pathDataAn <- paste(sharedPath, "dataAn/", sep = "")


for(k in 1:nrow(metadata)){
cmd <- paste("cp ",
             metadata$FilePath[k],
             " ",
             pathDataAn, 
             metadata$ShortName[k], ".", metadata$DataType[k], ".tsv",
             sep = "")
system(cmd)
}

metadata$DataAnFilePath <- pathDataAn
metadata$DataAnFileName <- paste(metadata$ShortName, ".", metadata$DataType, ".tsv", sep = "")

metadataMaster <- dcast(data = metadata,
                        Primer + RunNum + DataAnFilePath + ShortName
                        ~ DataType,
                        value.var = "DataAnFileName",
                        FUN       = c)

write.table(metadataMaster, 
            file      = paste(sharedPath, "My_MetadataTable2.tsv", sep = ""), 
            append    = FALSE,
            quote     = FALSE,
            sep       = "\t", 
            row.names = FALSE)

```
tax.fill to remove uncertae sedis, unclassified...

```{r}
library("RAM")

taxFillDir <- "taxFillTables/"
dir.create(paste(pathDataAn, taxFillDir, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
taxFillPath <- paste(pathDataAn, taxFillDir, sep = "")

i <- 1
#for(i in 1:nrow(metadataMaster)){
for(i in 1:nrow(metadataMaster)){
  temp <-  read.table(paste(metadataMaster$DataAnFilePath[i], metadataMaster$table[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, colClasses=c("taxonomy"="character"))
  temp2 <- temp
  temp2$row.names <- NULL #to unactivate numbers generated in the margin
  temp2 <- tax.fill(temp2, downstream=TRUE)
  temp$taxonomy <- temp2$taxonomy
  
  write.table(temp, file=paste(pathDataAn, "taxFillTables/", metadataMaster$ShortName[i], ".table.taxFill.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)
}  
 
```
Add 2 columns to the metadata file with the tax.fill tables names and paths (steps 1-6)
1-make a list of the filenames

```{r}
library("reshape2")
taxList <- list.files(path = taxFillPath, pattern = glob2rx("*taxFill.tsv*"), full.names=TRUE)
```
2-Tell R it is a dataframe

```{r}
metaTax <- data.frame(taxList)
```
3-Give header name to the column being built

```{r}
colnames(metaTax)[1] <- "taxPath"
```
4-Build a 2nd colum with file name only 

```{r}
metaTax$taxName     <- basename(taxList)
```
5-Cut basename from the path name

```{r}
metaTax$taxPath <- sub('([^//]+\\//).*', '\\1', metaTax$taxPath)
#Here, we match all characters that are not ? ([^?]+) followed by ? (\\?) and use parentheses to capture as a group (([^?]+\\?)), and we leave the rest of characters not in the group (./*). (http://stackoverflow.com/questions/10150579/adding-a-column-to-a-data-frame)
```
6-Remove last /
```{r}
metaTax$taxPath <- substr(metaTax$taxPath, 1, nchar(metaTax$taxPath)-1) 
```
Add-up your two new columns (taxPath and taxName) to the end of the metadata file
```{r}
metadataMaster2$taxPath     <- metaTax$taxPath
metadataMaster2$taxName     <- metaTax$taxName
```
write the precious table you just made 
```{r}
write.table(metadataMasterTax, 
            file      = paste(sharedPath, "metadataMasterTax.tsv", sep = ""), 
            append    = FALSE,
            quote     = FALSE,
            sep       = "\t", 
            row.names = FALSE)
```

Create diversity directory and path. List your datasets for diversity indexes. Adds a bunch of columns in a new diversity metadata file.
```{r}
diversityDir <- "diversityDir/"
dir.create(paste(pathDataAn, diversityDir, sep = ""),
           showWarnings = TRUE,
           recursive    = FALSE)
diversityPath <- paste(pathDataAn, diversityDir, sep = "")


i <- 1
for(i in 1:nrow(metadataMasterTax)) {
  temp <- read.table(paste(metadataMasterTax$DataAnFilePath[i], metadataMasterTax$meta[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE)
 
  temp$row.names <- NULL

  trans <-  read.table(paste(metadataMasterTax$taxPath[i], metadataMasterTax$taxName[i], sep = ""),
                      sep = "\t", header = TRUE, dec = ".", comment.char = "", quote = "", stringsAsFactors = TRUE,
                      as.is = TRUE, check.names = FALSE)
  
  trans$row.names <- NULL
  
  rownames(temp) <- colnames(trans)[-ncol(trans)]
   
  temp2 <- OTU.diversity(data = list(data = trans), meta = temp)
  write.table(temp2, file=paste(pathDataAn, "diversityDir/", metadataMasterTax$ShortName[i], ".meta.div.tsv", sep = ""),
              append    = FALSE,
              sep       = "\t",
              row.names = FALSE,
              quote=FALSE)
  }

```


```{r}


```